<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Moosa Rezwani" />

<meta name="date" content="2025-02-07" />

<title>2.B: JASPAR &amp; rbioapi</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>
<style type="text/css">
a.anchor-section {margin-left: 10px; visibility: hidden; color: inherit;}
.hasAnchor:hover a.anchor-section {visibility: visible;}
ul > li > .anchor-section {display: none;}
</style>
<style type="text/css">
a.anchor-section::before {content: '#';font-size: 80%;}
</style>
<script>document.addEventListener('DOMContentLoaded', function () {
  // If section divs is used, we need to put the anchor in the child header
  const headers = document.querySelectorAll("div.hasAnchor.section[class*='level'] > :first-child")

  headers.forEach(function (x) {
    // Add to the header node
    if (!x.classList.contains('hasAnchor')) x.classList.add('hasAnchor')
    // Remove from the section or div created by Pandoc
    x.parentElement.classList.remove('hasAnchor')
  })
})
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">2.B: JASPAR &amp; rbioapi</h1>
<h4 class="author">Moosa Rezwani</h4>
<h4 class="date">2025-02-07</h4>


<div id="TOC">
<ul>
<li><a href="#introduction" id="toc-introduction"><span class="toc-section-number">1</span> Introduction</a></li>
<li><a href="#data-organization-in-jaspar" id="toc-data-organization-in-jaspar"><span class="toc-section-number">2</span> Data Organization in JASPAR</a>
<ul>
<li><a href="#releases" id="toc-releases"><span class="toc-section-number">2.1</span> Releases</a></li>
<li><a href="#collections" id="toc-collections"><span class="toc-section-number">2.2</span> Collections</a></li>
<li><a href="#taxonomic-groups" id="toc-taxonomic-groups"><span class="toc-section-number">2.3</span> Taxonomic Groups</a></li>
<li><a href="#species" id="toc-species"><span class="toc-section-number">2.4</span> Species</a></li>
</ul></li>
<li><a href="#matrix-profiles" id="toc-matrix-profiles"><span class="toc-section-number">3</span> Matrix Profiles</a>
<ul>
<li><a href="#search-matrix-profiles" id="toc-search-matrix-profiles"><span class="toc-section-number">3.1</span> Search Matrix Profiles</a></li>
<li><a href="#list-matrix-profiles-associated-to-a-base-identifier" id="toc-list-matrix-profiles-associated-to-a-base-identifier"><span class="toc-section-number">3.2</span> List Matrix Profiles Associated to
a Base identifier</a></li>
<li><a href="#get-a-matrix-profile" id="toc-get-a-matrix-profile"><span class="toc-section-number">3.3</span> Get a Matrix Profile</a>
<ul>
<li><a href="#get-matrix-and-annotations-as-an-r-object" id="toc-get-matrix-and-annotations-as-an-r-object"><span class="toc-section-number">3.3.1</span> Get Matrix and Annotations as an
R Object</a></li>
<li><a href="#save-a-matrix-a-file-in-specific-format" id="toc-save-a-matrix-a-file-in-specific-format"><span class="toc-section-number">3.3.2</span> Save a Matrix a File in Specific
Format</a></li>
</ul></li>
<li><a href="#get-binding-sites-of-a-matrix-profiles" id="toc-get-binding-sites-of-a-matrix-profiles"><span class="toc-section-number">3.4</span> Get Binding Sites of a Matrix
Profiles</a></li>
</ul></li>
<li><a href="#tf-flexible-models-tffms" id="toc-tf-flexible-models-tffms"><span class="toc-section-number">4</span> TF flexible models (TFFMs)</a></li>
<li><a href="#citations" id="toc-citations"><span class="toc-section-number">5</span> How to Cite?</a></li>
<li><a href="#links" id="toc-links"><span class="toc-section-number">6</span> Links</a></li>
<li><a href="#session-info" id="toc-session-info"><span class="toc-section-number">7</span> Session info</a></li>
</ul>
</div>

<div id="introduction" class="section level1 hasAnchor" number="1">
<h1 class="hasAnchor"><span class="header-section-number">1</span>
Introduction<a href="#introduction" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Directly quoting from Fornes O, Castro-Mondragon JA, Khan A, et
al:</p>
<blockquote>
<p>JASPAR (<a href="https://jaspar.elixir.no">https://jaspar.elixir.no)</a> is an
open-access database of curated, non-redundant transcription factor
(TF)-binding profiles stored as position frequency matrices (PFMs) for
TFs across multiple species in six taxonomic groups. In this 8th release
of JASPAR, the CORE collection has been expanded with 245 new PFMs (169
for vertebrates, 42 for plants, 17 for nematodes, 10 for insects, and 7
for fungi), and 156 PFMs were updated (125 for vertebrates, 28 for
plants and 3 for insects). These new profiles represent an 18% expansion
compared to the previous release.</p>
<p>source:<br />
Fornes O, Castro-Mondragon JA, Khan A, et al. JASPAR 2020: update of the
open-access database of transcription factor binding profiles.
<em>Nucleic Acids Res</em>. 2019; doi: <a href="https://doi.org/10.1093/nar/gkz1001">10.1093/nar/gkz1001</a></p>
</blockquote>
<hr />
</div>
<div id="data-organization-in-jaspar" class="section level1 hasAnchor" number="2">
<h1 class="hasAnchor"><span class="header-section-number">2</span> Data
Organization in JASPAR<a href="#data-organization-in-jaspar" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>JASPAR is a database of transcription factor binding matrices with
annotations and metadata. These entities are organized in a hierarchical
fashion that we will explore next.</p>
<div id="releases" class="section level2 hasAnchor" number="2.1">
<h2 class="hasAnchor"><span class="header-section-number">2.1</span>
Releases<a href="#releases" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In addition to the latest JASPAR database release (2020), other
active releases are also available. Most of the rbioapi JASPAR functions
have a <code>release</code> argument that allows you to use other
database releases.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="do">## Call the function without any arguments to get a list of releases</span></span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a>releases <span class="ot">&lt;-</span> <span class="fu">rba_jaspar_releases</span>()</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a></span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a><span class="do">## Supply a release number for details:</span></span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a>release_7_info <span class="ot">&lt;-</span> <span class="fu">rba_jaspar_releases</span>(<span class="dv">7</span>)</span></code></pre></div>
</div>
<div id="collections" class="section level2 hasAnchor" number="2.2">
<h2 class="hasAnchor"><span class="header-section-number">2.2</span>
Collections<a href="#collections" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Within a release, Matrix profiles are organized into collections, You
can use <code>rba_jaspar_collections()</code> to get a list of available
collections, or read “JASPAR Collections” section in <a href="https://jaspar.elixir.no/docs/" title="JASPAR Documentation">documentation page in JASPAR web-site</a>
for a thorough review.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="do">## To get a list of available collection in release 2020:</span></span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="fu">rba_jaspar_collections</span>(<span class="at">release =</span> <span class="dv">2020</span>)</span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a><span class="co">#&gt;          name                                                      url</span></span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a><span class="co">#&gt; 1        CORE        https://jaspar.elixir.no/api/v1/collections/CORE/</span></span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a><span class="co">#&gt; 2 UNVALIDATED https://jaspar.elixir.no/api/v1/collections/UNVALIDATED/</span></span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a></span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a></span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a><span class="do">## You can list information of all matrices available in a collection:</span></span>
<span id="cb2-9"><a href="#cb2-9" tabindex="-1"></a>mat_in_core_2020 <span class="ot">&lt;-</span> <span class="fu">rba_jaspar_collections_matrices</span>(<span class="at">collection =</span> <span class="st">&quot;CORE&quot;</span>)</span></code></pre></div>
</div>
<div id="taxonomic-groups" class="section level2 hasAnchor" number="2.3">
<h2 class="hasAnchor"><span class="header-section-number">2.3</span>
Taxonomic Groups<a href="#taxonomic-groups" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Within each collection, the matrix profiles are organized based on
main taxonomic groups:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="do">## To get a list of taxonomic groups in release 2020:</span></span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a><span class="fu">rba_jaspar_taxons</span>(<span class="at">release =</span> <span class="dv">2020</span>)</span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a><span class="co">#&gt;             name                                                  url</span></span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a><span class="co">#&gt; 1         plants        https://jaspar.elixir.no/api/v1/taxon/plants/</span></span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a><span class="co">#&gt; 2    vertebrates   https://jaspar.elixir.no/api/v1/taxon/vertebrates/</span></span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a><span class="co">#&gt; 3        insects       https://jaspar.elixir.no/api/v1/taxon/insects/</span></span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a><span class="co">#&gt; 4   urochordates  https://jaspar.elixir.no/api/v1/taxon/urochordates/</span></span>
<span id="cb3-8"><a href="#cb3-8" tabindex="-1"></a><span class="co">#&gt; 5      nematodes     https://jaspar.elixir.no/api/v1/taxon/nematodes/</span></span>
<span id="cb3-9"><a href="#cb3-9" tabindex="-1"></a><span class="co">#&gt; 6          fungi         https://jaspar.elixir.no/api/v1/taxon/fungi/</span></span>
<span id="cb3-10"><a href="#cb3-10" tabindex="-1"></a><span class="co">#&gt; 7        diatoms       https://jaspar.elixir.no/api/v1/taxon/diatoms/</span></span>
<span id="cb3-11"><a href="#cb3-11" tabindex="-1"></a><span class="co">#&gt; 8     trematodes    https://jaspar.elixir.no/api/v1/taxon/trematodes/</span></span>
<span id="cb3-12"><a href="#cb3-12" tabindex="-1"></a><span class="co">#&gt; 9  dictyostelium https://jaspar.elixir.no/api/v1/taxon/dictyostelium/</span></span>
<span id="cb3-13"><a href="#cb3-13" tabindex="-1"></a><span class="co">#&gt; 10      cnidaria      https://jaspar.elixir.no/api/v1/taxon/cnidaria/</span></span>
<span id="cb3-14"><a href="#cb3-14" tabindex="-1"></a><span class="co">#&gt; 11      oomycota      https://jaspar.elixir.no/api/v1/taxon/oomycota/</span></span>
<span id="cb3-15"><a href="#cb3-15" tabindex="-1"></a></span>
<span id="cb3-16"><a href="#cb3-16" tabindex="-1"></a></span>
<span id="cb3-17"><a href="#cb3-17" tabindex="-1"></a><span class="do">## You can list information of all matrices available in a taxonomic group:</span></span>
<span id="cb3-18"><a href="#cb3-18" tabindex="-1"></a>mat_in_insects <span class="ot">&lt;-</span> <span class="fu">rba_jaspar_taxons_matrices</span>(<span class="at">tax_group =</span> <span class="st">&quot;insects&quot;</span>)</span></code></pre></div>
</div>
<div id="species" class="section level2 hasAnchor" number="2.4">
<h2 class="hasAnchor"><span class="header-section-number">2.4</span>
Species<a href="#species" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>As we go down in the data organization hierarchy, Each taxonomic
group consist of species:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="do">## To get a list of species in release 2020:</span></span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a>species <span class="ot">&lt;-</span> <span class="fu">rba_jaspar_species</span>(<span class="at">release =</span> <span class="dv">2020</span>)</span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a><span class="fu">head</span>(species)</span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a><span class="co">#&gt;   tax_id                          species</span></span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a><span class="co">#&gt; 1   5037           Ajellomyces capsulatus</span></span>
<span id="cb4-6"><a href="#cb4-6" tabindex="-1"></a><span class="co">#&gt; 2   4151                Antirrhinum majus</span></span>
<span id="cb4-7"><a href="#cb4-7" tabindex="-1"></a><span class="co">#&gt; 3  81972 Arabidopsis lyrata subsp. lyrata</span></span>
<span id="cb4-8"><a href="#cb4-8" tabindex="-1"></a><span class="co">#&gt; 4   3702             Arabidopsis thaliana</span></span>
<span id="cb4-9"><a href="#cb4-9" tabindex="-1"></a><span class="co">#&gt; 5   9913                       Bos taurus</span></span>
<span id="cb4-10"><a href="#cb4-10" tabindex="-1"></a><span class="co">#&gt; 6   6238          Caenorhabditis briggsae</span></span>
<span id="cb4-11"><a href="#cb4-11" tabindex="-1"></a><span class="co">#&gt;                                              url</span></span>
<span id="cb4-12"><a href="#cb4-12" tabindex="-1"></a><span class="co">#&gt; 1  https://jaspar.elixir.no/api/v1/species/5037/</span></span>
<span id="cb4-13"><a href="#cb4-13" tabindex="-1"></a><span class="co">#&gt; 2  https://jaspar.elixir.no/api/v1/species/4151/</span></span>
<span id="cb4-14"><a href="#cb4-14" tabindex="-1"></a><span class="co">#&gt; 3 https://jaspar.elixir.no/api/v1/species/81972/</span></span>
<span id="cb4-15"><a href="#cb4-15" tabindex="-1"></a><span class="co">#&gt; 4  https://jaspar.elixir.no/api/v1/species/3702/</span></span>
<span id="cb4-16"><a href="#cb4-16" tabindex="-1"></a><span class="co">#&gt; 5  https://jaspar.elixir.no/api/v1/species/9913/</span></span>
<span id="cb4-17"><a href="#cb4-17" tabindex="-1"></a><span class="co">#&gt; 6  https://jaspar.elixir.no/api/v1/species/6238/</span></span>
<span id="cb4-18"><a href="#cb4-18" tabindex="-1"></a><span class="co">#&gt;                                       matrix_url</span></span>
<span id="cb4-19"><a href="#cb4-19" tabindex="-1"></a><span class="co">#&gt; 1  https://jaspar.elixir.no/api/v1/species/5037/</span></span>
<span id="cb4-20"><a href="#cb4-20" tabindex="-1"></a><span class="co">#&gt; 2  https://jaspar.elixir.no/api/v1/species/4151/</span></span>
<span id="cb4-21"><a href="#cb4-21" tabindex="-1"></a><span class="co">#&gt; 3 https://jaspar.elixir.no/api/v1/species/81972/</span></span>
<span id="cb4-22"><a href="#cb4-22" tabindex="-1"></a><span class="co">#&gt; 4  https://jaspar.elixir.no/api/v1/species/3702/</span></span>
<span id="cb4-23"><a href="#cb4-23" tabindex="-1"></a><span class="co">#&gt; 5  https://jaspar.elixir.no/api/v1/species/9913/</span></span>
<span id="cb4-24"><a href="#cb4-24" tabindex="-1"></a><span class="co">#&gt; 6  https://jaspar.elixir.no/api/v1/species/6238/</span></span>
<span id="cb4-25"><a href="#cb4-25" tabindex="-1"></a></span>
<span id="cb4-26"><a href="#cb4-26" tabindex="-1"></a><span class="do">## You can list information of all matrices available in a specie:</span></span>
<span id="cb4-27"><a href="#cb4-27" tabindex="-1"></a>mat_in_human <span class="ot">&lt;-</span> <span class="fu">rba_jaspar_species_matrices</span>(<span class="at">tax_id =</span> <span class="dv">9606</span>)</span></code></pre></div>
<hr />
</div>
</div>
<div id="matrix-profiles" class="section level1 hasAnchor" number="3">
<h1 class="hasAnchor"><span class="header-section-number">3</span>
Matrix Profiles<a href="#matrix-profiles" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="search-matrix-profiles" class="section level2 hasAnchor" number="3.1">
<h2 class="hasAnchor"><span class="header-section-number">3.1</span>
Search Matrix Profiles<a href="#search-matrix-profiles" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Retrieving a list of every matrix available in a given category is
not the only option. You can also build a search query using
<code>rba_jaspar_matrix_search</code>. Note that this is a search
function, you are not required to fill every argument. You may use any
combination of arguments you see fit to build your query. You can even
call the function without any argument to get a list of all the matrix
profiles. For instance:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="do">## Get a list of all the available matrix profile:</span></span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a>all_matrices <span class="ot">&lt;-</span> <span class="fu">rba_jaspar_matrix_search</span>()</span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a></span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a><span class="do">## Search FOX:</span></span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a>FOX_matrices <span class="ot">&lt;-</span> <span class="fu">rba_jaspar_matrix_search</span>(<span class="at">term =</span> <span class="st">&quot;FOX&quot;</span>)</span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a></span>
<span id="cb5-7"><a href="#cb5-7" tabindex="-1"></a><span class="do">## Transcription factors named FOXP3</span></span>
<span id="cb5-8"><a href="#cb5-8" tabindex="-1"></a>FOXP3_matrices <span class="ot">&lt;-</span> <span class="fu">rba_jaspar_matrix_search</span>(<span class="at">term =</span> <span class="st">&quot;FOXP3&quot;</span>)</span>
<span id="cb5-9"><a href="#cb5-9" tabindex="-1"></a></span>
<span id="cb5-10"><a href="#cb5-10" tabindex="-1"></a><span class="do">## Transcription factors of Zipper-Type Class</span></span>
<span id="cb5-11"><a href="#cb5-11" tabindex="-1"></a>zipper_matrices <span class="ot">&lt;-</span> <span class="fu">rba_jaspar_matrix_search</span>(<span class="at">tf_class =</span> <span class="st">&quot;Zipper-Type&quot;</span>)</span>
<span id="cb5-12"><a href="#cb5-12" tabindex="-1"></a></span>
<span id="cb5-13"><a href="#cb5-13" tabindex="-1"></a><span class="do">## Transcription factors of Zipper-Type Class in PBM collection</span></span>
<span id="cb5-14"><a href="#cb5-14" tabindex="-1"></a>zipper_pbm_matrices <span class="ot">&lt;-</span> <span class="fu">rba_jaspar_matrix_search</span>(</span>
<span id="cb5-15"><a href="#cb5-15" tabindex="-1"></a>  <span class="at">tf_class =</span> <span class="st">&quot;Zipper-Type&quot;</span>,</span>
<span id="cb5-16"><a href="#cb5-16" tabindex="-1"></a>  <span class="at">collection =</span> <span class="st">&quot;PBM&quot;</span></span>
<span id="cb5-17"><a href="#cb5-17" tabindex="-1"></a>)</span></code></pre></div>
</div>
<div id="list-matrix-profiles-associated-to-a-base-identifier" class="section level2 hasAnchor" number="3.2">
<h2 class="hasAnchor"><span class="header-section-number">3.2</span>
List Matrix Profiles Associated to a Base identifier<a href="#list-matrix-profiles-associated-to-a-base-identifier" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Since JASPAR release 2010, the matrix profiles are versioned. A
matrix profile Identifier has a “base_id.version” naming schema; for
example “MA0600.2” corresponds to the second version of a matrix with
base ID MA0600. You can Use <code>rba_jaspar_matrix_versions</code> to
get a list of matrix profiles with a given base ID. Also note that some
functions, generally those that are used to list available matrices,
have an argument called <code>only_last_version</code>.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="do">## Get matrix profiles versions associated to a base id</span></span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a>MA0600_versions <span class="ot">&lt;-</span> <span class="fu">rba_jaspar_matrix_versions</span>(<span class="st">&quot;MA0600&quot;</span>)</span></code></pre></div>
</div>
<div id="get-a-matrix-profile" class="section level2 hasAnchor" number="3.3">
<h2 class="hasAnchor"><span class="header-section-number">3.3</span> Get
a Matrix Profile<a href="#get-a-matrix-profile" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Now that you listed or searched for matrix profiles, you can use
<code>rba_jaspar_matrix</code> to retrieve matrix profiles. There are
two ways in which you can use this function:</p>
<div id="get-matrix-and-annotations-as-an-r-object" class="section level3 hasAnchor" number="3.3.1">
<h3 class="hasAnchor"><span class="header-section-number">3.3.1</span>
Get Matrix and Annotations as an R Object<a href="#get-matrix-and-annotations-as-an-r-object" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>To do that, only fill in the <code>matrix_id</code> argument in
<code>rba_jaspar_matrix</code></p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a>pfm_matrix <span class="ot">&lt;-</span> <span class="fu">rba_jaspar_matrix</span>(<span class="at">matrix_id =</span> <span class="st">&quot;MA0600.2&quot;</span>)</span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a></span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a><span class="do">## you can find the matrix in the pfm element along with</span></span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a><span class="do">## other elements which correspond to annotations and details</span></span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a><span class="fu">str</span>(pfm_matrix)</span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a><span class="co">#&gt; List of 24</span></span>
<span id="cb7-7"><a href="#cb7-7" tabindex="-1"></a><span class="co">#&gt;  $ matrix_id    : chr &quot;MA0600.2&quot;</span></span>
<span id="cb7-8"><a href="#cb7-8" tabindex="-1"></a><span class="co">#&gt;  $ name         : chr &quot;RFX2&quot;</span></span>
<span id="cb7-9"><a href="#cb7-9" tabindex="-1"></a><span class="co">#&gt;  $ base_id      : chr &quot;MA0600&quot;</span></span>
<span id="cb7-10"><a href="#cb7-10" tabindex="-1"></a><span class="co">#&gt;  $ version      : int 2</span></span>
<span id="cb7-11"><a href="#cb7-11" tabindex="-1"></a><span class="co">#&gt;  $ collection   : chr &quot;CORE&quot;</span></span>
<span id="cb7-12"><a href="#cb7-12" tabindex="-1"></a><span class="co">#&gt;  $ sequence_logo: chr &quot;https://jaspar.elixir.no/static/logos/svg/MA0600.2.svg&quot;</span></span>
<span id="cb7-13"><a href="#cb7-13" tabindex="-1"></a><span class="co">#&gt;  $ versions_url : chr &quot;https://jaspar.elixir.no/api/v1/matrix/MA0600/versions&quot;</span></span>
<span id="cb7-14"><a href="#cb7-14" tabindex="-1"></a><span class="co">#&gt;  $ sites_url    : NULL</span></span>
<span id="cb7-15"><a href="#cb7-15" tabindex="-1"></a><span class="co">#&gt;  $ pfm          : num [1:4, 1:16] 1381 5653 4042 2336 270 ...</span></span>
<span id="cb7-16"><a href="#cb7-16" tabindex="-1"></a><span class="co">#&gt;   ..- attr(*, &quot;dimnames&quot;)=List of 2</span></span>
<span id="cb7-17"><a href="#cb7-17" tabindex="-1"></a><span class="co">#&gt;   .. ..$ : chr [1:4] &quot;A&quot; &quot;C&quot; &quot;G&quot; &quot;T&quot;</span></span>
<span id="cb7-18"><a href="#cb7-18" tabindex="-1"></a><span class="co">#&gt;   .. ..$ : NULL</span></span>
<span id="cb7-19"><a href="#cb7-19" tabindex="-1"></a><span class="co">#&gt;  $ class        : chr &quot;Fork head/winged helix factors&quot;</span></span>
<span id="cb7-20"><a href="#cb7-20" tabindex="-1"></a><span class="co">#&gt;  $ family       : chr &quot;RFX-related factors&quot;</span></span>
<span id="cb7-21"><a href="#cb7-21" tabindex="-1"></a><span class="co">#&gt;  $ tfe_id       : list()</span></span>
<span id="cb7-22"><a href="#cb7-22" tabindex="-1"></a><span class="co">#&gt;  $ medline      : chr &quot;8754849&quot;</span></span>
<span id="cb7-23"><a href="#cb7-23" tabindex="-1"></a><span class="co">#&gt;  $ pazar_tf_id  : list()</span></span>
<span id="cb7-24"><a href="#cb7-24" tabindex="-1"></a><span class="co">#&gt;  $ remap_tf_name: chr &quot;RFX2&quot;</span></span>
<span id="cb7-25"><a href="#cb7-25" tabindex="-1"></a><span class="co">#&gt;  $ source       : chr &quot;23332764&quot;</span></span>
<span id="cb7-26"><a href="#cb7-26" tabindex="-1"></a><span class="co">#&gt;  $ tax_group    : chr &quot;vertebrates&quot;</span></span>
<span id="cb7-27"><a href="#cb7-27" tabindex="-1"></a><span class="co">#&gt;  $ type         : chr &quot;HT-SELEX&quot;</span></span>
<span id="cb7-28"><a href="#cb7-28" tabindex="-1"></a><span class="co">#&gt;  $ tfe_ids      : list()</span></span>
<span id="cb7-29"><a href="#cb7-29" tabindex="-1"></a><span class="co">#&gt;  $ pubmed_ids   : chr &quot;8754849&quot;</span></span>
<span id="cb7-30"><a href="#cb7-30" tabindex="-1"></a><span class="co">#&gt;  $ pazar_tf_ids : list()</span></span>
<span id="cb7-31"><a href="#cb7-31" tabindex="-1"></a><span class="co">#&gt;  $ uniprot_ids  : chr &quot;P48378&quot;</span></span>
<span id="cb7-32"><a href="#cb7-32" tabindex="-1"></a><span class="co">#&gt;  $ species      :&#39;data.frame&#39;:   1 obs. of  2 variables:</span></span>
<span id="cb7-33"><a href="#cb7-33" tabindex="-1"></a><span class="co">#&gt;   ..$ tax_id: int 9606</span></span>
<span id="cb7-34"><a href="#cb7-34" tabindex="-1"></a><span class="co">#&gt;   ..$ name  : chr &quot;Homo sapiens&quot;</span></span>
<span id="cb7-35"><a href="#cb7-35" tabindex="-1"></a><span class="co">#&gt;  $ tffm         :List of 7</span></span>
<span id="cb7-36"><a href="#cb7-36" tabindex="-1"></a><span class="co">#&gt;   ..$ tffm_id        : chr &quot;TFFM0576.1&quot;</span></span>
<span id="cb7-37"><a href="#cb7-37" tabindex="-1"></a><span class="co">#&gt;   ..$ base_id        : chr &quot;TFFM0576&quot;</span></span>
<span id="cb7-38"><a href="#cb7-38" tabindex="-1"></a><span class="co">#&gt;   ..$ version        : int 1</span></span>
<span id="cb7-39"><a href="#cb7-39" tabindex="-1"></a><span class="co">#&gt;   ..$ log_p_1st_order: num 6275</span></span>
<span id="cb7-40"><a href="#cb7-40" tabindex="-1"></a><span class="co">#&gt;   ..$ log_p_detailed : num 6660</span></span>
<span id="cb7-41"><a href="#cb7-41" tabindex="-1"></a><span class="co">#&gt;   ..$ experiment_name: chr &quot;CistromeDB_58298&quot;</span></span>
<span id="cb7-42"><a href="#cb7-42" tabindex="-1"></a><span class="co">#&gt;   ..$ tffm_url       : chr &quot;https://jaspar.elixir.no/api/v1/tffm/TFFM0576.1/&quot;</span></span></code></pre></div>
</div>
<div id="save-a-matrix-a-file-in-specific-format" class="section level3 hasAnchor" number="3.3.2">
<h3 class="hasAnchor"><span class="header-section-number">3.3.2</span>
Save a Matrix a File in Specific Format<a href="#save-a-matrix-a-file-in-specific-format" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>JASPAR provides position frequency matrices (PFM) formatted as
<strong>Raw PFM</strong>, <strong>JASPAR</strong>,
<strong>TRANSFAC</strong>, <strong>YAML</strong>, and
<strong>MEME</strong>. You can download a matrix profile as a file with
any of these formats. To do that, You should use the
<code>file_format</code> and <code>save_to</code> arguments available in
<code>rba_jaspar_matrix</code>. There are two notes here:</p>
<ol style="list-style-type: decimal">
<li><p>In this case, the function will save your matrix as a file and
returns the un-parsed content of the file as a character
string.</p></li>
<li><p>The <code>save_to</code> argument in this function, and in fact
through any rbioapi function can be used in many ways:<br />
2.1. save_to = NA: rbioapi will automatically generate a file path under
your working directory, save the file in that path , and informs you
with a message.<br />
2.2 save_to = file_name without path: rbioapi will save the file with
your supplied name in your working directory.<br />
2.3. save_to = a directory path (without file): rbioapi will save the
file with a proper name in that directory.<br />
2.4. save_to = a file path (i.e. ending with .extension): rbioapi will
save the file exactly to this path. Make sure that the file extension of
the path matches your requested file format. If this was not the case,
rbioapi will save the file with the extension supplied in the path, but
issues a warning to inform you about that.</p>
<p>In any of the aforementioned cases, the file path can be absolute or
relative.</p></li>
</ol>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="do">## Different wqays in which you can save the matrix file:</span></span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a>meme_matrix1 <span class="ot">&lt;-</span> <span class="fu">rba_jaspar_matrix</span>(</span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a>  <span class="at">matrix_id =</span> <span class="st">&quot;MA0600.2&quot;</span>,</span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a>  <span class="at">file_format =</span> <span class="st">&quot;meme&quot;</span></span>
<span id="cb8-5"><a href="#cb8-5" tabindex="-1"></a>)</span>
<span id="cb8-6"><a href="#cb8-6" tabindex="-1"></a></span>
<span id="cb8-7"><a href="#cb8-7" tabindex="-1"></a>meme_matrix2 <span class="ot">&lt;-</span> <span class="fu">rba_jaspar_matrix</span>(</span>
<span id="cb8-8"><a href="#cb8-8" tabindex="-1"></a>  <span class="at">matrix_id =</span> <span class="st">&quot;MA0600.2&quot;</span>,</span>
<span id="cb8-9"><a href="#cb8-9" tabindex="-1"></a>  <span class="at">file_format =</span> <span class="st">&quot;meme&quot;</span>,</span>
<span id="cb8-10"><a href="#cb8-10" tabindex="-1"></a>  <span class="at">save_to =</span> <span class="st">&quot;my_matrix.meme&quot;</span></span>
<span id="cb8-11"><a href="#cb8-11" tabindex="-1"></a>)</span>
<span id="cb8-12"><a href="#cb8-12" tabindex="-1"></a></span>
<span id="cb8-13"><a href="#cb8-13" tabindex="-1"></a>meme_matrix3 <span class="ot">&lt;-</span> <span class="fu">rba_jaspar_matrix</span>(</span>
<span id="cb8-14"><a href="#cb8-14" tabindex="-1"></a>  <span class="at">matrix_id =</span> <span class="st">&quot;MA0600.2&quot;</span>,</span>
<span id="cb8-15"><a href="#cb8-15" tabindex="-1"></a>  <span class="at">file_format =</span> <span class="st">&quot;meme&quot;</span>,</span>
<span id="cb8-16"><a href="#cb8-16" tabindex="-1"></a>  <span class="at">save_to =</span> <span class="st">&quot;c:/rbioapi&quot;</span></span>
<span id="cb8-17"><a href="#cb8-17" tabindex="-1"></a>)</span>
<span id="cb8-18"><a href="#cb8-18" tabindex="-1"></a></span>
<span id="cb8-19"><a href="#cb8-19" tabindex="-1"></a>meme_matrix4 <span class="ot">&lt;-</span> <span class="fu">rba_jaspar_matrix</span>(</span>
<span id="cb8-20"><a href="#cb8-20" tabindex="-1"></a>  <span class="at">matrix_id =</span> <span class="st">&quot;MA0600.2&quot;</span>,</span>
<span id="cb8-21"><a href="#cb8-21" tabindex="-1"></a>  <span class="at">file_format =</span> <span class="st">&quot;meme&quot;</span>,</span>
<span id="cb8-22"><a href="#cb8-22" tabindex="-1"></a>  <span class="at">save_to =</span> <span class="st">&quot;c:/rbioapi/my_matrix.meme&quot;</span></span>
<span id="cb8-23"><a href="#cb8-23" tabindex="-1"></a>)</span></code></pre></div>
</div>
</div>
<div id="get-binding-sites-of-a-matrix-profiles" class="section level2 hasAnchor" number="3.4">
<h2 class="hasAnchor"><span class="header-section-number">3.4</span> Get
Binding Sites of a Matrix Profiles<a href="#get-binding-sites-of-a-matrix-profiles" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>If available, you can retrieve information on binding sites
associated with a matrix profile. The information includes a data frame
of genomic coordination of the binding site, URL to FASTA and BED files,
along with other annotations.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a><span class="do">## Get binding site of a matrix profile:</span></span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a>binding_sites <span class="ot">&lt;-</span> <span class="fu">rba_jaspar_sites</span>(<span class="at">matrix_id =</span> <span class="st">&quot;MA0600.2&quot;</span>)</span></code></pre></div>
<hr />
</div>
</div>
<div id="tf-flexible-models-tffms" class="section level1 hasAnchor" number="4">
<h1 class="hasAnchor"><span class="header-section-number">4</span> TF
flexible models (TFFMs)<a href="#tf-flexible-models-tffms" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>JASPAR also stores and assigns identifiers to TF flexible models
(TFFMs). Just like PFM (position frequency matrices), you can search
TFFMs or retrieve information and annotations using a TFFM Identifier.
TFFM IDs are versioned, meaning that they are in base_id.version
format.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a><span class="do">## Search TFFMs. This is a search function. Thus, what has been presented</span></span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a><span class="do">## in `Search Matrix Profiles` section also applies here:</span></span>
<span id="cb10-3"><a href="#cb10-3" tabindex="-1"></a></span>
<span id="cb10-4"><a href="#cb10-4" tabindex="-1"></a><span class="do">## Get a list of all the available matrix profile:</span></span>
<span id="cb10-5"><a href="#cb10-5" tabindex="-1"></a>all_tffms <span class="ot">&lt;-</span> <span class="fu">rba_jaspar_tffm_search</span>()</span>
<span id="cb10-6"><a href="#cb10-6" tabindex="-1"></a></span>
<span id="cb10-7"><a href="#cb10-7" tabindex="-1"></a><span class="do">## Search FOX:</span></span>
<span id="cb10-8"><a href="#cb10-8" tabindex="-1"></a>FOX_tffms <span class="ot">&lt;-</span> <span class="fu">rba_jaspar_tffm_search</span>(<span class="at">term =</span> <span class="st">&quot;FOX&quot;</span>)</span>
<span id="cb10-9"><a href="#cb10-9" tabindex="-1"></a></span>
<span id="cb10-10"><a href="#cb10-10" tabindex="-1"></a><span class="do">## Transcription factors named FOXP3</span></span>
<span id="cb10-11"><a href="#cb10-11" tabindex="-1"></a>FOXP3_tffms <span class="ot">&lt;-</span> <span class="fu">rba_jaspar_tffm_search</span>(<span class="at">term =</span> <span class="st">&quot;FOXP3&quot;</span>)</span>
<span id="cb10-12"><a href="#cb10-12" tabindex="-1"></a></span>
<span id="cb10-13"><a href="#cb10-13" tabindex="-1"></a><span class="do">## Transcription factors of insects taxonomic group</span></span>
<span id="cb10-14"><a href="#cb10-14" tabindex="-1"></a>insects_tffms <span class="ot">&lt;-</span> <span class="fu">rba_jaspar_tffm_search</span>(<span class="at">tax_group =</span> <span class="st">&quot;insects&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a><span class="do">## Now that you have a TFFM ID, you can retrieve it</span></span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a>TFFM0056 <span class="ot">&lt;-</span> <span class="fu">rba_jaspar_tffm</span>(<span class="st">&quot;TFFM0056.3&quot;</span>)</span>
<span id="cb11-3"><a href="#cb11-3" tabindex="-1"></a></span>
<span id="cb11-4"><a href="#cb11-4" tabindex="-1"></a><span class="fu">str</span>(TFFM0056)</span>
<span id="cb11-5"><a href="#cb11-5" tabindex="-1"></a><span class="co">#&gt; List of 10</span></span>
<span id="cb11-6"><a href="#cb11-6" tabindex="-1"></a><span class="co">#&gt;  $ tffm_id        : chr &quot;TFFM0056.3&quot;</span></span>
<span id="cb11-7"><a href="#cb11-7" tabindex="-1"></a><span class="co">#&gt;  $ base_id        : chr &quot;TFFM0056&quot;</span></span>
<span id="cb11-8"><a href="#cb11-8" tabindex="-1"></a><span class="co">#&gt;  $ version        : int 3</span></span>
<span id="cb11-9"><a href="#cb11-9" tabindex="-1"></a><span class="co">#&gt;  $ matrix_base_id : chr &quot;MA0039&quot;</span></span>
<span id="cb11-10"><a href="#cb11-10" tabindex="-1"></a><span class="co">#&gt;  $ matrix_id      : chr &quot;MA0039.4&quot;</span></span>
<span id="cb11-11"><a href="#cb11-11" tabindex="-1"></a><span class="co">#&gt;  $ matrix_url     : chr &quot;https://jaspar.elixir.no/api/v1/matrix/MA0039.4/&quot;</span></span>
<span id="cb11-12"><a href="#cb11-12" tabindex="-1"></a><span class="co">#&gt;  $ matrix_version : int 4</span></span>
<span id="cb11-13"><a href="#cb11-13" tabindex="-1"></a><span class="co">#&gt;  $ experiment_name: chr &quot;CistromeDB_33718&quot;</span></span>
<span id="cb11-14"><a href="#cb11-14" tabindex="-1"></a><span class="co">#&gt;  $ first_order    :List of 5</span></span>
<span id="cb11-15"><a href="#cb11-15" tabindex="-1"></a><span class="co">#&gt;   ..$ log_p       : num 7420</span></span>
<span id="cb11-16"><a href="#cb11-16" tabindex="-1"></a><span class="co">#&gt;   ..$ dense_logo  : chr &quot;https://jaspar.elixir.no/static/TFFM/TFFM0056.3/TFFM_first_order_trained_dense_logo.svg&quot;</span></span>
<span id="cb11-17"><a href="#cb11-17" tabindex="-1"></a><span class="co">#&gt;   ..$ summary_logo: chr &quot;https://jaspar.elixir.no/static/TFFM/TFFM0056.3/TFFM_first_order_trained_summary_logo.svg&quot;</span></span>
<span id="cb11-18"><a href="#cb11-18" tabindex="-1"></a><span class="co">#&gt;   ..$ hits        : chr &quot;https://jaspar.elixir.no/static/TFFM/TFFM0056.3/TFFM_first_order_trained.hits.svg&quot;</span></span>
<span id="cb11-19"><a href="#cb11-19" tabindex="-1"></a><span class="co">#&gt;   ..$ xml         : chr &quot;https://jaspar.elixir.no/static/TFFM/TFFM0056.3/TFFM_first_order_trained.xml&quot;</span></span>
<span id="cb11-20"><a href="#cb11-20" tabindex="-1"></a><span class="co">#&gt;  $ detailed       :List of 5</span></span>
<span id="cb11-21"><a href="#cb11-21" tabindex="-1"></a><span class="co">#&gt;   ..$ log_p       : num 6854</span></span>
<span id="cb11-22"><a href="#cb11-22" tabindex="-1"></a><span class="co">#&gt;   ..$ dense_logo  : chr &quot;https://jaspar.elixir.no/static/TFFM/TFFM0056.3/TFFM_detailed_trained_dense_logo.svg&quot;</span></span>
<span id="cb11-23"><a href="#cb11-23" tabindex="-1"></a><span class="co">#&gt;   ..$ summary_logo: chr &quot;https://jaspar.elixir.no/static/TFFM/TFFM0056.3/TFFM_detailed_trained_summary_logo.svg&quot;</span></span>
<span id="cb11-24"><a href="#cb11-24" tabindex="-1"></a><span class="co">#&gt;   ..$ hits        : chr &quot;https://jaspar.elixir.no/static/TFFM/TFFM0056.3/TFFM_detailed_trained.hits.svg&quot;</span></span>
<span id="cb11-25"><a href="#cb11-25" tabindex="-1"></a><span class="co">#&gt;   ..$ xml         : chr &quot;https://jaspar.elixir.no/static/TFFM/TFFM0056.3/TFFM_detailed_trained.xml&quot;</span></span></code></pre></div>
<hr />
</div>
<div id="citations" class="section level1 hasAnchor" number="5">
<h1 class="hasAnchor"><span class="header-section-number">5</span> How
to Cite?<a href="#citations" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>To cite JASPAR (Please see <a href="https://jaspar.elixir.no/faq/" class="uri">https://jaspar.elixir.no/faq/</a>):</p>
<ul>
<li>Rauluseviciute I, Riudavets-Puig R, Blanc-Mathieu R,
Castro-Mondragon JA, Ferenc K, Kumar V, Lemma RB, Lucas J, Chèneby J,
Baranasic D, Khan A, Fornes O, Gundersen S, Johansen M, Hovig E, Lenhard
B, Sandelin A, Wasserman WW, Parcy F, Mathelier A <em>JASPAR 2024: 20th
anniversary of the open-access database of transcription factor binding
profiles</em> Nucleic Acids Res. in_press; <a href="https://doi.org/10.1093/nar/gkad1059" class="uri">https://doi.org/10.1093/nar/gkad1059</a></li>
<li>Khan, A. and Mathelier, A. <em>JASPAR RESTful API: accessing JASPAR
data from any programming language</em>. Bioinformatics, 2017, <a href="https://doi.org/10.1093/bioinformatics/btx804" class="uri">https://doi.org/10.1093/bioinformatics/btx804</a></li>
</ul>
<p>To cite rbioapi:</p>
<ul>
<li>Moosa Rezwani, Ali Akbar Pourfathollah, Farshid Noorbakhsh, rbioapi:
user-friendly R interface to biologic web services’ API, Bioinformatics,
Volume 38, Issue 10, 15 May 2022, Pages 2952–2953, <a href="https://doi.org/10.1093/bioinformatics/btac172" class="uri">https://doi.org/10.1093/bioinformatics/btac172</a></li>
</ul>
<hr />
</div>
<div id="links" class="section level1 hasAnchor" number="6">
<h1 class="hasAnchor"><span class="header-section-number">6</span>
Links<a href="#links" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<ul>
<li><p><a href="https://rbioapi.moosa-r.com/articles/rbioapi_jaspar.html" title="2.G: JASPAR &amp; rbioapi">This article in rbioapi
documentation site</a></p></li>
<li><p><a href="https://rbioapi.moosa-r.com/reference/index.html#section-jaspar-rba-jaspar-" title="rbioapi reference">Functions references in rbioapi
documentation site</a></p></li>
<li><p><a href="rbioapi.html" title="rbioapi: User-Friendly R Interface to Biologic Web Services&#39; API">rbioapi
vignette index</a></p></li>
</ul>
<hr />
</div>
<div id="session-info" class="section level1 hasAnchor" number="7">
<h1 class="hasAnchor"><span class="header-section-number">7</span>
Session info<a href="#session-info" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<pre><code>#&gt; R version 4.4.2 (2024-10-31 ucrt)
#&gt; Platform: x86_64-w64-mingw32/x64
#&gt; Running under: Windows 11 x64 (build 26100)
#&gt; 
#&gt; Matrix products: default
#&gt; 
#&gt; 
#&gt; locale:
#&gt; [1] LC_COLLATE=C                          
#&gt; [2] LC_CTYPE=English_United States.utf8   
#&gt; [3] LC_MONETARY=English_United States.utf8
#&gt; [4] LC_NUMERIC=C                          
#&gt; [5] LC_TIME=English_United States.utf8    
#&gt; 
#&gt; time zone: Europe/Brussels
#&gt; tzcode source: internal
#&gt; 
#&gt; attached base packages:
#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     
#&gt; 
#&gt; other attached packages:
#&gt; [1] rbioapi_0.8.2
#&gt; 
#&gt; loaded via a namespace (and not attached):
#&gt;  [1] digest_0.6.37     R6_2.5.1          fastmap_1.2.0     xfun_0.50        
#&gt;  [5] magrittr_2.0.3    cachem_1.1.0      knitr_1.49        htmltools_0.5.8.1
#&gt;  [9] png_0.1-8         rmarkdown_2.29    lifecycle_1.0.4   DT_0.33          
#&gt; [13] cli_3.6.3         grid_4.4.2        sass_0.4.9        jquerylib_0.1.4  
#&gt; [17] compiler_4.4.2    httr_1.4.7        rstudioapi_0.17.1 tools_4.4.2      
#&gt; [21] curl_6.1.0        evaluate_1.0.3    bslib_0.8.0       yaml_2.3.10      
#&gt; [25] htmlwidgets_1.6.4 rlang_1.1.4       jsonlite_1.8.9    crosstalk_1.2.1</code></pre>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
