<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Moosa Rezwani" />

<meta name="date" content="2022-04-05" />

<title>2.B: JASPAR &amp; rbioapi</title>

<script src="data:application/javascript;base64,Ly8gUGFuZG9jIDIuOSBhZGRzIGF0dHJpYnV0ZXMgb24gYm90aCBoZWFkZXIgYW5kIGRpdi4gV2UgcmVtb3ZlIHRoZSBmb3JtZXIgKHRvCi8vIGJlIGNvbXBhdGlibGUgd2l0aCB0aGUgYmVoYXZpb3Igb2YgUGFuZG9jIDwgMi44KS4KZG9jdW1lbnQuYWRkRXZlbnRMaXN0ZW5lcignRE9NQ29udGVudExvYWRlZCcsIGZ1bmN0aW9uKGUpIHsKICB2YXIgaHMgPSBkb2N1bWVudC5xdWVyeVNlbGVjdG9yQWxsKCJkaXYuc2VjdGlvbltjbGFzcyo9J2xldmVsJ10gPiA6Zmlyc3QtY2hpbGQiKTsKICB2YXIgaSwgaCwgYTsKICBmb3IgKGkgPSAwOyBpIDwgaHMubGVuZ3RoOyBpKyspIHsKICAgIGggPSBoc1tpXTsKICAgIGlmICghL15oWzEtNl0kL2kudGVzdChoLnRhZ05hbWUpKSBjb250aW51ZTsgIC8vIGl0IHNob3VsZCBiZSBhIGhlYWRlciBoMS1oNgogICAgYSA9IGguYXR0cmlidXRlczsKICAgIHdoaWxlIChhLmxlbmd0aCA+IDApIGgucmVtb3ZlQXR0cmlidXRlKGFbMF0ubmFtZSk7CiAgfQp9KTsK"></script>
<link href="data:text/css,%0Aa%2Eanchor%2Dsection%20%7Bmargin%2Dleft%3A%2010px%3B%20visibility%3A%20hidden%3B%20color%3A%20inherit%3B%7D%0A%2EhasAnchor%3Ahover%20a%2Eanchor%2Dsection%20%7Bvisibility%3A%20visible%3B%7D%0Aul%20%3E%20li%20%3E%20%2Eanchor%2Dsection%20%7Bdisplay%3A%20none%3B%7D%0A" rel="stylesheet" />
<link href="data:text/css,%0Aa%2Eanchor%2Dsection%3A%3Abefore%20%7Bcontent%3A%20%27%23%27%3Bfont%2Dsize%3A%2080%25%3B%7D%0A" rel="stylesheet" />
<script src="data:application/javascript;base64,ZG9jdW1lbnQuYWRkRXZlbnRMaXN0ZW5lcignRE9NQ29udGVudExvYWRlZCcsIGZ1bmN0aW9uICgpIHsKICAvLyBJZiBzZWN0aW9uIGRpdnMgaXMgdXNlZCwgd2UgbmVlZCB0byBwdXQgdGhlIGFuY2hvciBpbiB0aGUgY2hpbGQgaGVhZGVyCiAgY29uc3QgaGVhZGVycyA9IGRvY3VtZW50LnF1ZXJ5U2VsZWN0b3JBbGwoImRpdi5oYXNBbmNob3Iuc2VjdGlvbltjbGFzcyo9J2xldmVsJ10gPiA6Zmlyc3QtY2hpbGQiKQoKICBoZWFkZXJzLmZvckVhY2goZnVuY3Rpb24gKHgpIHsKICAgIC8vIEFkZCB0byB0aGUgaGVhZGVyIG5vZGUKICAgIGlmICgheC5jbGFzc0xpc3QuY29udGFpbnMoJ2hhc0FuY2hvcicpKSB4LmNsYXNzTGlzdC5hZGQoJ2hhc0FuY2hvcicpCiAgICAvLyBSZW1vdmUgZnJvbSB0aGUgc2VjdGlvbiBvciBkaXYgY3JlYXRlZCBieSBQYW5kb2MKICAgIHgucGFyZW50RWxlbWVudC5jbGFzc0xpc3QucmVtb3ZlKCdoYXNBbmNob3InKQogIH0pCn0pCg=="></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<link rel="stylesheet" href="data:text/css,body%20%7B%0Abackground%2Dcolor%3A%20%23fff%3B%0Amargin%3A%201em%20auto%3B%0Amax%2Dwidth%3A%20700px%3B%0Aoverflow%3A%20visible%3B%0Apadding%2Dleft%3A%202em%3B%0Apadding%2Dright%3A%202em%3B%0Afont%2Dfamily%3A%20%22Open%20Sans%22%2C%20%22Helvetica%20Neue%22%2C%20Helvetica%2C%20Arial%2C%20sans%2Dserif%3B%0Afont%2Dsize%3A%2014px%3B%0Aline%2Dheight%3A%201%2E35%3B%0A%7D%0A%23TOC%20%7B%0Aclear%3A%20both%3B%0Amargin%3A%200%200%2010px%2010px%3B%0Apadding%3A%204px%3B%0Awidth%3A%20400px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Aborder%2Dradius%3A%205px%3B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Afont%2Dsize%3A%2013px%3B%0Aline%2Dheight%3A%201%2E3%3B%0A%7D%0A%23TOC%20%2Etoctitle%20%7B%0Afont%2Dweight%3A%20bold%3B%0Afont%2Dsize%3A%2015px%3B%0Amargin%2Dleft%3A%205px%3B%0A%7D%0A%23TOC%20ul%20%7B%0Apadding%2Dleft%3A%2040px%3B%0Amargin%2Dleft%3A%20%2D1%2E5em%3B%0Amargin%2Dtop%3A%205px%3B%0Amargin%2Dbottom%3A%205px%3B%0A%7D%0A%23TOC%20ul%20ul%20%7B%0Amargin%2Dleft%3A%20%2D2em%3B%0A%7D%0A%23TOC%20li%20%7B%0Aline%2Dheight%3A%2016px%3B%0A%7D%0Atable%20%7B%0Amargin%3A%201em%20auto%3B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dcolor%3A%20%23DDDDDD%3B%0Aborder%2Dstyle%3A%20outset%3B%0Aborder%2Dcollapse%3A%20collapse%3B%0A%7D%0Atable%20th%20%7B%0Aborder%2Dwidth%3A%202px%3B%0Apadding%3A%205px%3B%0Aborder%2Dstyle%3A%20inset%3B%0A%7D%0Atable%20td%20%7B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dstyle%3A%20inset%3B%0Aline%2Dheight%3A%2018px%3B%0Apadding%3A%205px%205px%3B%0A%7D%0Atable%2C%20table%20th%2C%20table%20td%20%7B%0Aborder%2Dleft%2Dstyle%3A%20none%3B%0Aborder%2Dright%2Dstyle%3A%20none%3B%0A%7D%0Atable%20thead%2C%20table%20tr%2Eeven%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Ap%20%7B%0Amargin%3A%200%2E5em%200%3B%0A%7D%0Ablockquote%20%7B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Apadding%3A%200%2E25em%200%2E75em%3B%0A%7D%0Ahr%20%7B%0Aborder%2Dstyle%3A%20solid%3B%0Aborder%3A%20none%3B%0Aborder%2Dtop%3A%201px%20solid%20%23777%3B%0Amargin%3A%2028px%200%3B%0A%7D%0Adl%20%7B%0Amargin%2Dleft%3A%200%3B%0A%7D%0Adl%20dd%20%7B%0Amargin%2Dbottom%3A%2013px%3B%0Amargin%2Dleft%3A%2013px%3B%0A%7D%0Adl%20dt%20%7B%0Afont%2Dweight%3A%20bold%3B%0A%7D%0Aul%20%7B%0Amargin%2Dtop%3A%200%3B%0A%7D%0Aul%20li%20%7B%0Alist%2Dstyle%3A%20circle%20outside%3B%0A%7D%0Aul%20ul%20%7B%0Amargin%2Dbottom%3A%200%3B%0A%7D%0Apre%2C%20code%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0Aborder%2Dradius%3A%203px%3B%0Acolor%3A%20%23333%3B%0Awhite%2Dspace%3A%20pre%2Dwrap%3B%20%0A%7D%0Apre%20%7B%0Aborder%2Dradius%3A%203px%3B%0Amargin%3A%205px%200px%2010px%200px%3B%0Apadding%3A%2010px%3B%0A%7D%0Apre%3Anot%28%5Bclass%5D%29%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Acode%20%7B%0Afont%2Dfamily%3A%20Consolas%2C%20Monaco%2C%20%27Courier%20New%27%2C%20monospace%3B%0Afont%2Dsize%3A%2085%25%3B%0A%7D%0Ap%20%3E%20code%2C%20li%20%3E%20code%20%7B%0Apadding%3A%202px%200px%3B%0A%7D%0Adiv%2Efigure%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0Aimg%20%7B%0Abackground%2Dcolor%3A%20%23FFFFFF%3B%0Apadding%3A%202px%3B%0Aborder%3A%201px%20solid%20%23DDDDDD%3B%0Aborder%2Dradius%3A%203px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Amargin%3A%200%205px%3B%0A%7D%0Ah1%20%7B%0Amargin%2Dtop%3A%200%3B%0Afont%2Dsize%3A%2035px%3B%0Aline%2Dheight%3A%2040px%3B%0A%7D%0Ah2%20%7B%0Aborder%2Dbottom%3A%204px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Apadding%2Dbottom%3A%202px%3B%0Afont%2Dsize%3A%20145%25%3B%0A%7D%0Ah3%20%7B%0Aborder%2Dbottom%3A%202px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Afont%2Dsize%3A%20120%25%3B%0A%7D%0Ah4%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23f7f7f7%3B%0Amargin%2Dleft%3A%208px%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Ah5%2C%20h6%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23ccc%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Aa%20%7B%0Acolor%3A%20%230033dd%3B%0Atext%2Ddecoration%3A%20none%3B%0A%7D%0Aa%3Ahover%20%7B%0Acolor%3A%20%236666ff%3B%20%7D%0Aa%3Avisited%20%7B%0Acolor%3A%20%23800080%3B%20%7D%0Aa%3Avisited%3Ahover%20%7B%0Acolor%3A%20%23BB00BB%3B%20%7D%0Aa%5Bhref%5E%3D%22http%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0Aa%5Bhref%5E%3D%22https%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0A%0Acode%20%3E%20span%2Ekw%20%7B%20color%3A%20%23555%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Edt%20%7B%20color%3A%20%23902000%3B%20%7D%20%0Acode%20%3E%20span%2Edv%20%7B%20color%3A%20%2340a070%3B%20%7D%20%0Acode%20%3E%20span%2Ebn%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Efl%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Ech%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Est%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Eco%20%7B%20color%3A%20%23888888%3B%20font%2Dstyle%3A%20italic%3B%20%7D%20%0Acode%20%3E%20span%2Eot%20%7B%20color%3A%20%23007020%3B%20%7D%20%0Acode%20%3E%20span%2Eal%20%7B%20color%3A%20%23ff0000%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Efu%20%7B%20color%3A%20%23900%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Eer%20%7B%20color%3A%20%23a61717%3B%20background%2Dcolor%3A%20%23e3d2d2%3B%20%7D%20%0A" type="text/css" />




</head>

<body>




<h1 class="title toc-ignore">2.B: JASPAR &amp; rbioapi</h1>
<h4 class="author">Moosa Rezwani</h4>
<h4 class="date">2022-04-05</h4>


<div id="TOC">
<ul>
<li><a href="#introduction"><span class="toc-section-number">1</span> Introduction<span></span></a></li>
<li><a href="#data-organization-in-jaspar"><span class="toc-section-number">2</span> Data Organization in JASPAR<span></span></a>
<ul>
<li><a href="#releases"><span class="toc-section-number">2.1</span> Releases<span></span></a></li>
<li><a href="#collections"><span class="toc-section-number">2.2</span> Collections<span></span></a></li>
<li><a href="#taxonomic-groups"><span class="toc-section-number">2.3</span> Taxonomic Groups<span></span></a></li>
<li><a href="#species"><span class="toc-section-number">2.4</span> Species<span></span></a></li>
</ul></li>
<li><a href="#matrix-profiles"><span class="toc-section-number">3</span> Matrix Profiles<span></span></a>
<ul>
<li><a href="#search-matrix-profiles"><span class="toc-section-number">3.1</span> Search Matrix Profiles<span></span></a></li>
<li><a href="#list-matrix-profiles-associated-to-a-base-identifier"><span class="toc-section-number">3.2</span> List Matrix Profiles Associated to a Base identifier<span></span></a></li>
<li><a href="#get-a-matrix-profile"><span class="toc-section-number">3.3</span> Get a Matrix Profile<span></span></a>
<ul>
<li><a href="#get-matrix-and-annotations-as-an-r-object"><span class="toc-section-number">3.3.1</span> Get Matrix and Annotations as an R Object<span></span></a></li>
<li><a href="#save-a-matrix-a-file-in-specific-format"><span class="toc-section-number">3.3.2</span> Save a Matrix a File in Specific Format<span></span></a></li>
</ul></li>
<li><a href="#get-binding-sites-of-a-matrix-profiles"><span class="toc-section-number">3.4</span> Get Binding Sites of a Matrix Profiles<span></span></a></li>
</ul></li>
<li><a href="#tf-flexible-models-tffms"><span class="toc-section-number">4</span> TF flexible models (TFFMs)<span></span></a></li>
<li><a href="#Citations"><span class="toc-section-number">5</span> How to Cite?<span></span></a></li>
<li><a href="#links"><span class="toc-section-number">6</span> Links<span></span></a></li>
<li><a href="#session-info"><span class="toc-section-number">7</span> Session info<span></span></a></li>
</ul>
</div>

<hr />
<div id="introduction" class="section level1 hasAnchor" number="1">
<h1 class="hasAnchor"><span class="header-section-number">1</span> Introduction<a href="#introduction" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Directly quoting from Fornes O, Castro-Mondragon JA, Khan A, et al:</p>
<blockquote>
<p>JASPAR (<a href="https://jaspar.genereg.net">https://jaspar.genereg.net)</a> is an open-access database of curated, non-redundant transcription factor (TF)-binding profiles stored as position frequency matrices (PFMs) for TFs across multiple species in six taxonomic groups. In this 8th release of JASPAR, the CORE collection has been expanded with 245 new PFMs (169 for vertebrates, 42 for plants, 17 for nematodes, 10 for insects, and 7 for fungi), and 156 PFMs were updated (125 for vertebrates, 28 for plants and 3 for insects). These new profiles represent an 18% expansion compared to the previous release.</p>
<p>source:<br />
Fornes O, Castro-Mondragon JA, Khan A, et al. JASPAR 2020: update of the open-access database of transcription factor binding profiles. <em>Nucleic Acids Res</em>. 2019; doi: <a href="https://doi.org/10.1093/nar/gkz1001">10.1093/nar/gkz1001</a></p>
</blockquote>
<hr />
</div>
<div id="data-organization-in-jaspar" class="section level1 hasAnchor" number="2">
<h1 class="hasAnchor"><span class="header-section-number">2</span> Data Organization in JASPAR<a href="#data-organization-in-jaspar" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>JASPAR is a database of transcription factor binding matrices with annotations and metadata. These entities are organized in a hierarchical fashion that we will explore next.</p>
<div id="releases" class="section level2 hasAnchor" number="2.1">
<h2 class="hasAnchor"><span class="header-section-number">2.1</span> Releases<a href="#releases" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In addition to the latest JASPAR database release (2020), other active releases are also available. Most of the rbioapi JASPAR functions have a <code>release</code> argument that allows you to use other database releases.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Call the function without any arguments to get a list of releases</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>releases <span class="ot">&lt;-</span> <span class="fu">rba_jaspar_releases</span>()</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="do">## Supply a release number for details:</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>release_7_info <span class="ot">&lt;-</span> <span class="fu">rba_jaspar_releases</span>(<span class="dv">7</span>)</span></code></pre></div>
</div>
<div id="collections" class="section level2 hasAnchor" number="2.2">
<h2 class="hasAnchor"><span class="header-section-number">2.2</span> Collections<a href="#collections" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Within a release, Matrix profiles are organized into collections, You can use <code>rba_jaspar_collections()</code> to get a list of available collections, or read “JASPAR Collections” section in <a href="https://jaspar.genereg.net/docs/" title="JASPAR Documentation">documentation page in JASPAR web-site</a> for a thorough review.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="do">## To get a list of available collection in release 2020:</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rba_jaspar_collections</span>(<span class="at">release =</span> <span class="dv">2020</span>)</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;                                                          url        name</span></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1        https://jaspar.genereg.net/api/v1/collections/CORE/        CORE</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2 https://jaspar.genereg.net/api/v1/collections/UNVALIDATED/ UNVALIDATED</span></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a><span class="do">## You can list information of all matrices available in a collection:</span></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>mat_in_core_2020 <span class="ot">&lt;-</span> <span class="fu">rba_jaspar_collections_matrices</span>(<span class="at">collection =</span> <span class="st">&quot;CORE&quot;</span>)</span></code></pre></div>
</div>
<div id="taxonomic-groups" class="section level2 hasAnchor" number="2.3">
<h2 class="hasAnchor"><span class="header-section-number">2.3</span> Taxonomic Groups<a href="#taxonomic-groups" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Within each collection, the matrix profiles are organized based on main taxonomic groups:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="do">## To get a list of taxonomic groups in release 2020:</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rba_jaspar_taxons</span>(<span class="at">release =</span> <span class="dv">2020</span>)</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;                                                       url          name</span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1         https://jaspar.genereg.net/api/v1/taxon/plants/        plants</span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2    https://jaspar.genereg.net/api/v1/taxon/vertebrates/   vertebrates</span></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3        https://jaspar.genereg.net/api/v1/taxon/insects/       insects</span></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4   https://jaspar.genereg.net/api/v1/taxon/urochordates/  urochordates</span></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5      https://jaspar.genereg.net/api/v1/taxon/nematodes/     nematodes</span></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 6          https://jaspar.genereg.net/api/v1/taxon/fungi/         fungi</span></span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 7     https://jaspar.genereg.net/api/v1/taxon/trematodes/    trematodes</span></span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 8  https://jaspar.genereg.net/api/v1/taxon/dictyostelium/ dictyostelium</span></span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 9       https://jaspar.genereg.net/api/v1/taxon/cnidaria/      cnidaria</span></span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 10      https://jaspar.genereg.net/api/v1/taxon/oomycota/      oomycota</span></span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a><span class="do">## You can list information of all matrices available in a taxonomic group:</span></span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a>mat_in_insects <span class="ot">&lt;-</span> <span class="fu">rba_jaspar_taxons_matrices</span>(<span class="at">tax_group =</span> <span class="st">&quot;insects&quot;</span>)</span></code></pre></div>
</div>
<div id="species" class="section level2 hasAnchor" number="2.4">
<h2 class="hasAnchor"><span class="header-section-number">2.4</span> Species<a href="#species" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>As we go down in the data organization hierarchy, Each taxonomic group consist of species:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="do">## To get a list of species in release 2020:</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>species <span class="ot">&lt;-</span> <span class="fu">rba_jaspar_species</span>(<span class="at">release =</span> <span class="dv">2020</span>)</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(species)</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   tax_id                          species</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1   5037           Ajellomyces capsulatus</span></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2   4151                Antirrhinum majus</span></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3  81972 Arabidopsis lyrata subsp. lyrata</span></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4   3702             Arabidopsis thaliana</span></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5   9913                       Bos taurus</span></span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 6   6238          Caenorhabditis briggsae</span></span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;                                                url</span></span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1  https://jaspar.genereg.net/api/v1/species/5037/</span></span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2  https://jaspar.genereg.net/api/v1/species/4151/</span></span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3 https://jaspar.genereg.net/api/v1/species/81972/</span></span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4  https://jaspar.genereg.net/api/v1/species/3702/</span></span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5  https://jaspar.genereg.net/api/v1/species/9913/</span></span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 6  https://jaspar.genereg.net/api/v1/species/6238/</span></span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;                                         matrix_url</span></span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1  https://jaspar.genereg.net/api/v1/species/5037/</span></span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2  https://jaspar.genereg.net/api/v1/species/4151/</span></span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3 https://jaspar.genereg.net/api/v1/species/81972/</span></span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4  https://jaspar.genereg.net/api/v1/species/3702/</span></span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5  https://jaspar.genereg.net/api/v1/species/9913/</span></span>
<span id="cb4-24"><a href="#cb4-24" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 6  https://jaspar.genereg.net/api/v1/species/6238/</span></span>
<span id="cb4-25"><a href="#cb4-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-26"><a href="#cb4-26" aria-hidden="true" tabindex="-1"></a><span class="do">## You can list information of all matrices available in a specie:</span></span>
<span id="cb4-27"><a href="#cb4-27" aria-hidden="true" tabindex="-1"></a>mat_in_human <span class="ot">&lt;-</span> <span class="fu">rba_jaspar_species_matrices</span>(<span class="at">tax_id =</span> <span class="dv">9606</span>)</span></code></pre></div>
<hr />
</div>
</div>
<div id="matrix-profiles" class="section level1 hasAnchor" number="3">
<h1 class="hasAnchor"><span class="header-section-number">3</span> Matrix Profiles<a href="#matrix-profiles" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="search-matrix-profiles" class="section level2 hasAnchor" number="3.1">
<h2 class="hasAnchor"><span class="header-section-number">3.1</span> Search Matrix Profiles<a href="#search-matrix-profiles" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Retrieving a list of every matrix available in a given category is not the only option. You can also build a search query using <code>rba_jaspar_matrix_search</code>. Note that this is a search function, you are not required to fill every argument. You may use any combination of arguments you see fit to build your query. You can even call the function without any argument to get a list of all the matrix profiles. For instance:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Get a list of all the available matrix profile:</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>all_matrices <span class="ot">&lt;-</span> <span class="fu">rba_jaspar_matrix_search</span>()</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="do">## Search FOX:</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>FOX_matrices <span class="ot">&lt;-</span> <span class="fu">rba_jaspar_matrix_search</span>(<span class="at">term =</span> <span class="st">&quot;FOX&quot;</span>)</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a><span class="do">## Transcription factors named FOXP3</span></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>FOXP3_matrices <span class="ot">&lt;-</span> <span class="fu">rba_jaspar_matrix_search</span>(<span class="at">term =</span> <span class="st">&quot;FOXP3&quot;</span>)</span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a><span class="do">## Transcription factors of Zipper-Type Class</span></span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>zipper_matrices <span class="ot">&lt;-</span> <span class="fu">rba_jaspar_matrix_search</span>(<span class="at">tf_class =</span> <span class="st">&quot;Zipper-Type&quot;</span>)</span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a><span class="do">## Transcription factors of Zipper-Type Class in PBM collection</span></span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a>zipper_pbm_matrices <span class="ot">&lt;-</span> <span class="fu">rba_jaspar_matrix_search</span>(<span class="at">tf_class =</span> <span class="st">&quot;Zipper-Type&quot;</span>,</span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a>                                                <span class="at">collection =</span> <span class="st">&quot;PBM&quot;</span>)</span></code></pre></div>
</div>
<div id="list-matrix-profiles-associated-to-a-base-identifier" class="section level2 hasAnchor" number="3.2">
<h2 class="hasAnchor"><span class="header-section-number">3.2</span> List Matrix Profiles Associated to a Base identifier<a href="#list-matrix-profiles-associated-to-a-base-identifier" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Since JASPAR release 2010, the matrix profiles are versioned. A matrix profile Identifier has a “base_id.version” naming schema; for example “MA0600.2” corresponds to the second version of a matrix with base ID MA0600. You can Use <code>rba_jaspar_matrix_versions</code> to get a list of matrix profiles with a given base ID. Also note that some functions, generally those that are used to list available matrices, have an argument called <code>only_last_version</code>.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Get matrix profiles versions associated to a base id</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>MA0600_versions <span class="ot">&lt;-</span> <span class="fu">rba_jaspar_matrix_versions</span>(<span class="st">&quot;MA0600&quot;</span>)</span></code></pre></div>
</div>
<div id="get-a-matrix-profile" class="section level2 hasAnchor" number="3.3">
<h2 class="hasAnchor"><span class="header-section-number">3.3</span> Get a Matrix Profile<a href="#get-a-matrix-profile" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Now that you listed or searched for matrix profiles, you can use <code>rba_jaspar_matrix</code> to retrieve matrix profiles. There are two ways in which you can use this function:</p>
<div id="get-matrix-and-annotations-as-an-r-object" class="section level3 hasAnchor" number="3.3.1">
<h3 class="hasAnchor"><span class="header-section-number">3.3.1</span> Get Matrix and Annotations as an R Object<a href="#get-matrix-and-annotations-as-an-r-object" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>To do that, only fill in the <code>matrix_id</code> argument in <code>rba_jaspar_matrix</code></p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>pfm_matrix <span class="ot">&lt;-</span> <span class="fu">rba_jaspar_matrix</span>(<span class="at">matrix_id =</span> <span class="st">&quot;MA0600.2&quot;</span>)</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="do">## you can find the matrix in the pfm element along with</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="do">## other elements which correspond to annotations and details</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(pfm_matrix)</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; List of 24</span></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  $ uniprot_ids  : list()</span></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  $ versions_url : chr &quot;https://jaspar.genereg.net/api/v1/matrix/MA0600/versions&quot;</span></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  $ tffm         : NULL</span></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  $ version      : int 2</span></span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  $ collection   : chr &quot;CORE&quot;</span></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  $ family       : chr &quot;RFX-related factors&quot;</span></span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  $ tfe_id       : list()</span></span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  $ medline      : chr &quot;8754849&quot;</span></span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  $ pubmed_ids   : chr &quot;8754849&quot;</span></span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  $ base_id      : chr &quot;MA0600&quot;</span></span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  $ sequence_logo: chr &quot;https://jaspar.genereg.net/static/logos/svg/MA0600.2.svg&quot;</span></span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  $ pazar_tf_ids : list()</span></span>
<span id="cb7-19"><a href="#cb7-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  $ name         : chr &quot;RFX2&quot;</span></span>
<span id="cb7-20"><a href="#cb7-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  $ source       : chr &quot;23332764&quot;</span></span>
<span id="cb7-21"><a href="#cb7-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  $ class        : chr &quot;Fork head/winged helix factors&quot;</span></span>
<span id="cb7-22"><a href="#cb7-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  $ matrix_id    : chr &quot;MA0600.2&quot;</span></span>
<span id="cb7-23"><a href="#cb7-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  $ tfe_ids      : list()</span></span>
<span id="cb7-24"><a href="#cb7-24" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  $ sites_url    : NULL</span></span>
<span id="cb7-25"><a href="#cb7-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  $ type         : chr &quot;HT-SELEX&quot;</span></span>
<span id="cb7-26"><a href="#cb7-26" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  $ pfm          : num [1:4, 1:16] 1381 5653 4042 2336 270 ...</span></span>
<span id="cb7-27"><a href="#cb7-27" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   ..- attr(*, &quot;dimnames&quot;)=List of 2</span></span>
<span id="cb7-28"><a href="#cb7-28" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   .. ..$ : chr [1:4] &quot;A&quot; &quot;C&quot; &quot;G&quot; &quot;T&quot;</span></span>
<span id="cb7-29"><a href="#cb7-29" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   .. ..$ : NULL</span></span>
<span id="cb7-30"><a href="#cb7-30" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  $ remap_tf_name: chr &quot;RFX2&quot;</span></span>
<span id="cb7-31"><a href="#cb7-31" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  $ tax_group    : chr &quot;vertebrates&quot;</span></span>
<span id="cb7-32"><a href="#cb7-32" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  $ species      :&#39;data.frame&#39;:   1 obs. of  2 variables:</span></span>
<span id="cb7-33"><a href="#cb7-33" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   ..$ name  : chr &quot;Homo sapiens&quot;</span></span>
<span id="cb7-34"><a href="#cb7-34" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   ..$ tax_id: int 9606</span></span>
<span id="cb7-35"><a href="#cb7-35" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  $ pazar_tf_id  : list()</span></span></code></pre></div>
</div>
<div id="save-a-matrix-a-file-in-specific-format" class="section level3 hasAnchor" number="3.3.2">
<h3 class="hasAnchor"><span class="header-section-number">3.3.2</span> Save a Matrix a File in Specific Format<a href="#save-a-matrix-a-file-in-specific-format" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>JASPAR provides position frequency matrices (PFM) formatted as <strong>Raw PFM</strong>, <strong>JASPAR</strong>, <strong>TRANSFAC</strong>, <strong>YAML</strong>, and <strong>MEME</strong>. You can download a matrix profile as a file with any of these formats. To do that, You should use the <code>file_format</code> and <code>save_to</code> arguments available in <code>rba_jaspar_matrix</code>. There are two notes here:</p>
<ol style="list-style-type: decimal">
<li><p>In this case, the function will save your matrix as a file and returns the un-parsed content of the file as a character string.</p></li>
<li><p>The <code>save_to</code> argument in this function, and in fact through any rbioapi function can be used in many ways:<br />
2.1. save_to = NA: rbioapi will automatically generate a file path under your working directory, save the file in that path , and informs you with a message.<br />
2.2 save_to = file_name without path: rbioapi will save the file with your supplied name in your working directory.<br />
2.3. save_to = a directory path (without file): rbioapi will save the file with a proper name in that directory.<br />
2.4. save_to = a file path (i.e. ending with .extension): rbioapi will save the file exactly to this path. Make sure that the file extension of the path matches your requested file format. If this was not the case, rbioapi will save the file with the extension supplied in the path, but issues a warning to inform you about that.</p>
<p>In any of the aforementioned cases, the file path can be absolute or relative.</p></li>
</ol>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Different wqays in which you can save the matrix file:</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>meme_matrix1 <span class="ot">&lt;-</span> <span class="fu">rba_jaspar_matrix</span>(<span class="at">matrix_id =</span> <span class="st">&quot;MA0600.2&quot;</span>,</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">file_format =</span> <span class="st">&quot;meme&quot;</span>)</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>meme_matrix2 <span class="ot">&lt;-</span> <span class="fu">rba_jaspar_matrix</span>(<span class="at">matrix_id =</span> <span class="st">&quot;MA0600.2&quot;</span>,</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">file_format =</span> <span class="st">&quot;meme&quot;</span>,</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">save_to =</span> <span class="st">&quot;my_matrix.meme&quot;</span>)</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>meme_matrix3 <span class="ot">&lt;-</span> <span class="fu">rba_jaspar_matrix</span>(<span class="at">matrix_id =</span> <span class="st">&quot;MA0600.2&quot;</span>,</span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">file_format =</span> <span class="st">&quot;meme&quot;</span>,</span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">save_to =</span> <span class="st">&quot;c:/rbioapi&quot;</span>)</span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a>meme_matrix4 <span class="ot">&lt;-</span> <span class="fu">rba_jaspar_matrix</span>(<span class="at">matrix_id =</span> <span class="st">&quot;MA0600.2&quot;</span>,</span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">file_format =</span> <span class="st">&quot;meme&quot;</span>,</span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">save_to =</span> <span class="st">&quot;c:/rbioapi/my_matrix.meme&quot;</span>)</span></code></pre></div>
</div>
</div>
<div id="get-binding-sites-of-a-matrix-profiles" class="section level2 hasAnchor" number="3.4">
<h2 class="hasAnchor"><span class="header-section-number">3.4</span> Get Binding Sites of a Matrix Profiles<a href="#get-binding-sites-of-a-matrix-profiles" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>If available, you can retrieve information on binding sites associated with a matrix profile. The information includes a data frame of genomic coordination of the binding site, URL to FASTA and BED files, along with other annotations.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Get binding site of a matrix profile:</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>binding_sites <span class="ot">&lt;-</span> <span class="fu">rba_jaspar_sites</span>(<span class="at">matrix_id =</span> <span class="st">&quot;MA0600.2&quot;</span>)</span></code></pre></div>
<hr />
</div>
</div>
<div id="tf-flexible-models-tffms" class="section level1 hasAnchor" number="4">
<h1 class="hasAnchor"><span class="header-section-number">4</span> TF flexible models (TFFMs)<a href="#tf-flexible-models-tffms" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>JASPAR also stores and assigns identifiers to TF flexible models (TFFMs). Just like PFM (position frequency matrices), you can search TFFMs or retrieve information and annotations using a TFFM Identifier. TFFM IDs are versioned, meaning that they are in base_id.version format.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Search TFFMs. This is a search function. Thus, what has been presented</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="do">## in `Search Matrix Profiles` section also applies here:</span></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a><span class="do">## Get a list of all the available matrix profile:</span></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>all_tffms <span class="ot">&lt;-</span> <span class="fu">rba_jaspar_tffm_search</span>()</span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a><span class="do">## Search FOX:</span></span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>FOX_tffms <span class="ot">&lt;-</span> <span class="fu">rba_jaspar_tffm_search</span>(<span class="at">term =</span> <span class="st">&quot;FOX&quot;</span>)</span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a><span class="do">## Transcription factors named FOXP3</span></span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a>FOXP3_tffms <span class="ot">&lt;-</span> <span class="fu">rba_jaspar_tffm_search</span>(<span class="at">term =</span> <span class="st">&quot;FOXP3&quot;</span>)</span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a><span class="do">## Transcription factors of insects taxonomic group</span></span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a>insects_tffms <span class="ot">&lt;-</span> <span class="fu">rba_jaspar_tffm_search</span>(<span class="at">tax_group =</span> <span class="st">&quot;insects&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Now that you have a TFFM ID, you can retrieve it</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>TFFM0056 <span class="ot">&lt;-</span> <span class="fu">rba_jaspar_tffm</span>(<span class="st">&quot;TFFM0056.3&quot;</span>)</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(TFFM0056)</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  Named list()</span></span></code></pre></div>
<hr />
</div>
<div id="Citations" class="section level1 hasAnchor" number="5">
<h1 class="hasAnchor"><span class="header-section-number">5</span> How to Cite?<a href="#Citations" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>To cite JASPAR (Please see <a href="https://jaspar.genereg.net/faq/" class="uri">https://jaspar.genereg.net/faq/</a>):</p>
<ul>
<li>Jaime A Castro-Mondragon, Rafael Riudavets-Puig, Ieva Rauluseviciute, Roza Berhanu Lemma, Laura Turchi, Romain Blanc-Mathieu, Jeremy Lucas, Paul Boddie, Aziz Khan, Nicolás Manosalva Pérez, Oriol Fornes, Tiffany Y Leung, Alejandro Aguirre, Fayrouz Hammal, Daniel Schmelter, Damir Baranasic, Benoit Ballester, Albin Sandelin, Boris Lenhard, Klaas Vandepoele, Wyeth W Wasserman, François Parcy, Anthony Mathelier, JASPAR 2022: the 9th release of the open-access database of transcription factor binding profiles, <em>Nucleic Acids Research</em>, Volume 50, Issue D1, 7 January 2022, Pages D165–D173, <a href="https://doi.org/10.1093/nar/gkab1113" class="uri">https://doi.org/10.1093/nar/gkab1113</a></li>
</ul>
<p>To cite rbioapi:</p>
<ul>
<li>Moosa Rezwani, Ali Akbar Pourfathollah, Farshid Noorbakhsh, rbioapi: User-Friendly R Interface to Biologic Web Services’ API, <em>Bioinformatics</em>, 2022;, btac172, <a href="https://doi.org/10.1093/bioinformatics/btac172" class="uri">https://doi.org/10.1093/bioinformatics/btac172</a></li>
</ul>
<hr />
</div>
<div id="links" class="section level1 heading2 hasAnchor" number="6">
<h1 class="heading2 hasAnchor"><span class="header-section-number">6</span> Links<a href="#links" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<ul>
<li><p><a href="https://rbioapi.moosa-r.com/articles/rbioapi_jaspar.html" title="2.G: JASPAR &amp; rbioapi">This article in rbioapi documentation site</a></p></li>
<li><p><a href="https://rbioapi.moosa-r.com/reference/index.html#section-jaspar-rba-jaspar-" title="rbioapi reference">Functions references in rbioapi documentation site</a></p></li>
<li><p><a href="rbioapi.html" title="rbioapi: User-Friendly R Interface to Biologic Web Services&#39; API">rbioapi vignette index</a></p></li>
</ul>
<hr />
</div>
<div id="session-info" class="section level1 heading2 hasAnchor" number="7">
<h1 class="heading2 hasAnchor"><span class="header-section-number">7</span> Session info<a href="#session-info" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<pre><code>#&gt; R version 4.1.2 (2021-11-01)
#&gt; Platform: x86_64-w64-mingw32/x64 (64-bit)
#&gt; Running under: Windows 10 x64 (build 19044)
#&gt; 
#&gt; Matrix products: default
#&gt; 
#&gt; locale:
#&gt; [1] LC_COLLATE=C                          
#&gt; [2] LC_CTYPE=English_United States.1252   
#&gt; [3] LC_MONETARY=English_United States.1252
#&gt; [4] LC_NUMERIC=C                          
#&gt; [5] LC_TIME=English_United States.1252    
#&gt; system code page: 1256
#&gt; 
#&gt; attached base packages:
#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     
#&gt; 
#&gt; other attached packages:
#&gt; [1] rbioapi_0.7.6
#&gt; 
#&gt; loaded via a namespace (and not attached):
#&gt;  [1] digest_0.6.29     R6_2.5.1          jsonlite_1.8.0    magrittr_2.0.2   
#&gt;  [5] evaluate_0.15     httr_1.4.2        stringi_1.7.6     rlang_1.0.2      
#&gt;  [9] cli_3.2.0         curl_4.3.2        rstudioapi_0.13   jquerylib_0.1.4  
#&gt; [13] DT_0.21           bslib_0.3.1       rmarkdown_2.13    tools_4.1.2      
#&gt; [17] stringr_1.4.0     htmlwidgets_1.5.4 crosstalk_1.2.0   xfun_0.30        
#&gt; [21] yaml_2.3.5        fastmap_1.1.0     compiler_4.1.2    htmltools_0.5.2  
#&gt; [25] knitr_1.37        sass_0.4.0</code></pre>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
