<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Moosa Rezwani" />

<meta name="date" content="2025-07-02" />

<title>1: rbioapi: User-Friendly R Interface to Biologic Web Services’ API</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>
<style type="text/css">
a.anchor-section {margin-left: 10px; visibility: hidden; color: inherit;}
.hasAnchor:hover a.anchor-section {visibility: visible;}
ul > li > .anchor-section {display: none;}
</style>
<style type="text/css">
a.anchor-section::before {content: '#';font-size: 80%;}
</style>
<script>document.addEventListener('DOMContentLoaded', function () {
  // If section divs is used, we need to put the anchor in the child header
  const headers = document.querySelectorAll("div.hasAnchor.section[class*='level'] > :first-child")

  headers.forEach(function (x) {
    // Add to the header node
    if (!x.classList.contains('hasAnchor')) x.classList.add('hasAnchor')
    // Remove from the section or div created by Pandoc
    x.parentElement.classList.remove('hasAnchor')
  })
})
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">1: rbioapi: User-Friendly R Interface to
Biologic Web Services’ API</h1>
<h4 class="author">Moosa Rezwani</h4>
<h4 class="date">2025-07-02</h4>


<div id="TOC">
<ul>
<li><a href="#what-does-rbioapi-do" id="toc-what-does-rbioapi-do"><span class="toc-section-number">1</span> What does rbioapi do?</a></li>
<li><a href="#what-is-supported-by-rbioapi" id="toc-what-is-supported-by-rbioapi"><span class="toc-section-number">2</span> What is Supported by
rbioapi?</a></li>
<li><a href="#How-to-install" id="toc-How-to-install"><span class="toc-section-number">3</span> How to install?</a></li>
<li><a href="#naming-conventions" id="toc-naming-conventions"><span class="toc-section-number">4</span> Naming conventions</a></li>
<li><a href="#changing-the-options" id="toc-changing-the-options"><span class="toc-section-number">5</span> Changing the options</a>
<ul>
<li><a href="#changing-the-options-change-the-option-globally" id="toc-changing-the-options-change-the-option-globally"><span class="toc-section-number">5.1</span> Change the option
globally</a></li>
<li><a href="#changing-the-options-change-the-option-only-within-a-function-call" id="toc-changing-the-options-change-the-option-only-within-a-function-call"><span class="toc-section-number">5.2</span> Change the option only within a
function call</a></li>
</ul></li>
<li><a href="#connection-test" id="toc-connection-test"><span class="toc-section-number">6</span> Connection test</a></li>
<li><a href="#multi-pages" id="toc-multi-pages"><span class="toc-section-number">7</span> Iterating over paginated
results</a></li>
<li><a href="#how-to-cite" id="toc-how-to-cite"><span class="toc-section-number">8</span> How and what to cite?</a>
<ul>
<li><a href="#how-to-cite-rbioapi" id="toc-how-to-cite-rbioapi"><span class="toc-section-number">8.1</span> How to cite rbioapi</a></li>
<li><a href="#how-to-cite-services" id="toc-how-to-cite-services"><span class="toc-section-number">8.2</span> How to cite the databases and web
services</a></li>
</ul></li>
<li><a href="#code-of-conduct" id="toc-code-of-conduct"><span class="toc-section-number">9</span> Code of conduct</a></li>
<li><a href="#what-next" id="toc-what-next"><span class="toc-section-number">10</span> What next?</a></li>
<li><a href="#design" id="toc-design"><span class="toc-section-number">11</span> Design of rbioapi</a></li>
<li><a href="#links" id="toc-links"><span class="toc-section-number">12</span> Links</a></li>
<li><a href="#session-info" id="toc-session-info"><span class="toc-section-number">13</span> Session info</a></li>
</ul>
</div>

<div id="what-does-rbioapi-do" class="section level1 hasAnchor" number="1">
<h1 class="hasAnchor"><span class="header-section-number">1</span> What
does rbioapi do?<a href="#what-does-rbioapi-do" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Currently fully supports <strong>Enrichr</strong>,
<strong>JASPAR</strong>, <strong>miEAA</strong>,
<strong>PANTHER</strong>, <strong>Reactome</strong>,
<strong>STRING</strong>, and <strong>UniProt</strong>!</p>
<p>The goal of rbioapi is to provide a user-friendly and consistent
interface to biological databases and services: In a way that insulates
the user from technicalities of using web services API and creates a
unified and easy-to-use interface to biological and medical web
services.</p>
<p>With rbioapi, you do not need to have technical knowledge about web
services API or learn how to work with a new package for every biologic
service or database. This an ongoing project; New databases and services
will be added periodically. Feel free to <a href="https://github.com/moosa-r/rbioapi/issues" title="Issue section in rbioapi GitHub repository">suggest</a> any
databases or services you often use.</p>
<hr />
</div>
<div id="what-is-supported-by-rbioapi" class="section level1 hasAnchor" number="2">
<h1 class="hasAnchor"><span class="header-section-number">2</span> What
is Supported by rbioapi?<a href="#what-is-supported-by-rbioapi" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>rbioapi is dedicated to <strong>Biological or Medical</strong>
databases and web services. Currently, rbioapi supports and covers every
API resources in the following services: (in alphabetical order):</p>
<p>On CRAN (Stable) version: (<a href="https://cran.r-project.org/package=rbioapi" class="uri">https://cran.r-project.org/package=rbioapi</a>)</p>
<ol style="list-style-type: decimal">
<li><a href="https://maayanlab.cloud/Enrichr/" title="Enrichr">Enrichr</a> (<a href="rbioapi_enrichr.html" title="rbioapi &amp; Enrichr vignette article">rbioapi vignette
article</a>) <sup>(new)</sup></li>
<li><a href="https://jaspar.elixir.no/" title="JASPAR - A database of transcription factor binding profiles">JASPAR</a>
(<a href="rbioapi_jaspar.html" title="rbioapi &amp; Enrichr vignette article">rbioapi vignette
article</a>) <sup>(new)</sup></li>
<li><a href="https://ccb-compute2.cs.uni-saarland.de/mieaa2" title="miRNA Enrichment Analysis and Annotation Tool (miEAA)">miEAA</a>
(<a href="rbioapi_mieaa.html" title="rbioapi &amp; miEAA vignette article">rbioapi vignette
article</a>)</li>
<li><a href="https://www.pantherdb.org" title="Protein Analysis THrough Evolutionary Relationships (PANTHER)">PANTHER</a>
(<a href="rbioapi_panther.html" title="rbioapi &amp; PANTHER vignette article">rbioapi vignette
article</a>)</li>
<li><a href="https://reactome.org/">Reactome</a> (<a href="rbioapi_reactome.html" title="rbioapi &amp; Reactome vignette article">rbioapi vignette
article</a>)</li>
<li><a href="https://string-db.org/" title="STRING: Protein-Protein Interaction Networks Functional Enrichment Analysis">STRING</a>
(<a href="rbioapi_string.html" title="rbioapi &amp; STRING vignette article">rbioapi vignette
article</a>)</li>
<li><a href="https://www.uniprot.org" title="Universal Protein Resource (UniProt)">UniProt</a> (<a href="rbioapi_uniprot.html" title="rbioapi &amp; UniProt vignette article">rbioapi vignette
article</a>)</li>
</ol>
<p>Only on Github (Developmental) version: (<a href="https://github.com/moosa-r/rbioapi/" class="uri">https://github.com/moosa-r/rbioapi/</a>):</p>
<ol style="list-style-type: decimal">
<li>currently none</li>
</ol>
<p>Each of the services has its dedicated vignette article. In this
article, I will write about the general framework of rbioapi. Make sure
to check the vignette article of each service to learn more about how to
use them.</p>
<p><strong>Note That:</strong> rbioapi is an ongoing project. New
databases and services will be implemented periodically in order to
gradually make the package as comprehensive as possible. Do you see
yourself often using a certain database/service? Feel free to suggest
any database/service by creating an issue on our GitHub <a href="https://github.com/moosa-r/" title="rbioapi GitHub repositry">repository</a>. I will appreciate any
suggestions.</p>
<hr />
</div>
<div id="How-to-install" class="section level1 hasAnchor" number="3">
<h1 class="hasAnchor"><span class="header-section-number">3</span> How
to install?<a href="#How-to-install" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>You can install the <a href="https://cran.r-project.org/package=rbioapi" title="rbioapi: User-Friendly R Interface to Biologic Web Services&#39; API">stable
release version of rbioapi</a> from <a href="https://cran.r-project.org/" title="The Comprehensive R Archive Network">CRAN</a> with:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;rbioapi&quot;</span>)</span></code></pre></div>
<p>However, the CRAN version is released at longer intervals, You can
install the most recent -development- version from <a href="https://github.com/moosa-r/rbioapi/" title="rbioapi repository on GitHub">GitHub</a> with:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;remotes&quot;</span>)</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a>remotes<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">&quot;moosa-r/rbioapi&quot;</span>)</span></code></pre></div>
<p>Now, we can load the package:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="fu">library</span>(rbioapi)</span></code></pre></div>
<hr />
</div>
<div id="naming-conventions" class="section level1 hasAnchor" number="4">
<h1 class="hasAnchor"><span class="header-section-number">4</span>
Naming conventions<a href="#naming-conventions" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>To keep the namespace organized, functions names follow this
pattern:</p>
<pre><code>rba_[service_name]_[resource_name]</code></pre>
<p>For example, <code>rba_string_version()</code> will call <a href="https://string-db.org/" title="STRING: Protein-Protein Interaction Networks Functional Enrichment Analysis">STRING</a>’s
version resource.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="fu">rba_string_version</span>()</span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a><span class="co">#&gt; Retrieving the STRING database version and address used by rbioapi.</span></span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a><span class="co">#&gt; $string_version</span></span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a><span class="co">#&gt; [1] &quot;12.0&quot;</span></span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a><span class="co">#&gt; $stable_address</span></span>
<span id="cb5-7"><a href="#cb5-7" tabindex="-1"></a><span class="co">#&gt; [1] &quot;https://version-12-0.string-db.org&quot;</span></span></code></pre></div>
<p>Thus, to this version, rbioapi function will have one of the
following naming schema:</p>
<ol style="list-style-type: decimal">
<li>rba_enrichr_*</li>
<li>rba_jaspar_*</li>
<li>rba_mieaa_*</li>
<li>rba_panther_*</li>
<li>rba_reactome_*</li>
<li>rba_string_*</li>
<li>rba_uniprot_*</li>
</ol>
<p>There are three exceptions: <code>rba_options()</code>,
<code>rba_connection_test()</code>, and <code>rba_pages()</code>; these
are helper functions. More on that later.</p>
<hr />
</div>
<div id="changing-the-options" class="section level1 hasAnchor" number="5">
<h1 class="hasAnchor"><span class="header-section-number">5</span>
Changing the options<a href="#changing-the-options" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>To give users greater control, rbioapi offers multiple configurable
options. See the manual of <code>rba_options()</code> function for a
full description of available options. In short, some of the options
will govern rbioapi’s connection with servers (e.g. timeout, retry) and
some of the options will modify your experience with rbioapi
(e.g. verbose, diagnostics, save_file). There are two ways that you may
use to change any option. Also, you can get table of available rbioapi
options and their current values by calling
<code>rba_options()</code>without any argument:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="fu">rba_options</span>()</span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a><span class="co">#&gt;   rbioapi_option current_value            allowed_value</span></span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a><span class="co">#&gt; 1    diagnostics         FALSE     Logical (TRUE/FALSE)</span></span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a><span class="co">#&gt; 2       dir_name       rbioapi                Character</span></span>
<span id="cb6-5"><a href="#cb6-5" tabindex="-1"></a><span class="co">#&gt; 3       progress         FALSE     Logical (TRUE/FALSE)</span></span>
<span id="cb6-6"><a href="#cb6-6" tabindex="-1"></a><span class="co">#&gt; 4      retry_max             0   Numeric (0 or greater)</span></span>
<span id="cb6-7"><a href="#cb6-7" tabindex="-1"></a><span class="co">#&gt; 5     retry_wait            10   Numeric (0 or greater)</span></span>
<span id="cb6-8"><a href="#cb6-8" tabindex="-1"></a><span class="co">#&gt; 6      save_file         FALSE     Logical (TRUE/FALSE)</span></span>
<span id="cb6-9"><a href="#cb6-9" tabindex="-1"></a><span class="co">#&gt; 7     skip_error          TRUE     Logical (TRUE/FALSE)</span></span>
<span id="cb6-10"><a href="#cb6-10" tabindex="-1"></a><span class="co">#&gt; 8        timeout            30 Numeric (0.1 or greater)</span></span>
<span id="cb6-11"><a href="#cb6-11" tabindex="-1"></a><span class="co">#&gt; 9        verbose          TRUE     Logical (TRUE/FALSE)</span></span></code></pre></div>
<p>Now, let us consider the ways in which we can alter the settings:</p>
<div id="changing-the-options-change-the-option-globally" class="section level2 hasAnchor" number="5.1">
<h2 class="hasAnchor"><span class="header-section-number">5.1</span>
Change the option globally<a href="#changing-the-options-change-the-option-globally" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Changing an option globally means that for the rest of your R
session, any rbioapi function will respect the changed option. To do
this, use <code>rba_options().</code> Each argument in this function
corresponds to a certain option; Thus by running this function with your
desired new values, you could globally alter that rbioapi option. for
example:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="fu">rba_options</span>(<span class="at">save_file =</span> <span class="cn">TRUE</span>)</span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a><span class="do">## From now on, the raw file of server&#39;s response will be saved to your working directory.</span></span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a></span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a><span class="fu">rba_options</span>(<span class="at">verbose =</span> <span class="cn">FALSE</span>)</span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a><span class="do">## From now on, the package will be quiet.</span></span></code></pre></div>
</div>
<div id="changing-the-options-change-the-option-only-within-a-function-call" class="section level2 hasAnchor" number="5.2">
<h2 class="hasAnchor"><span class="header-section-number">5.2</span>
Change the option only within a function call<a href="#changing-the-options-change-the-option-only-within-a-function-call" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>You can pass additional arguments to any rbioapi function using
“ellipsis” (the familiar <code>…</code> or dot dot dot!). Meaning that
you can call any function with additional arguments where each is
‘option = value’ pair. This way, any changes in options will be confined
within that particular function call. For example:</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="do">## Save the server&#39;s raw response file:</span></span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">rba_reactome_species</span>(</span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a>  <span class="at">only_main =</span> <span class="cn">TRUE</span>,</span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a>  <span class="at">save_file =</span> <span class="st">&quot;reactome_species.json&quot;</span></span>
<span id="cb8-5"><a href="#cb8-5" tabindex="-1"></a>)</span>
<span id="cb8-6"><a href="#cb8-6" tabindex="-1"></a></span>
<span id="cb8-7"><a href="#cb8-7" tabindex="-1"></a><span class="do">## Also, in the case of connection failure, retry up to 10 times:</span></span>
<span id="cb8-8"><a href="#cb8-8" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">rba_reactome_species</span>(</span>
<span id="cb8-9"><a href="#cb8-9" tabindex="-1"></a>  <span class="at">only_main =</span> <span class="cn">TRUE</span>,</span>
<span id="cb8-10"><a href="#cb8-10" tabindex="-1"></a>  <span class="at">save_file =</span> <span class="st">&quot;reactome_species.json&quot;</span>,</span>
<span id="cb8-11"><a href="#cb8-11" tabindex="-1"></a>  <span class="at">retry_max =</span> <span class="dv">10</span></span>
<span id="cb8-12"><a href="#cb8-12" tabindex="-1"></a>)</span></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a><span class="do">## Run these codes in your own R session to see the difference:</span></span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a></span>
<span id="cb9-3"><a href="#cb9-3" tabindex="-1"></a><span class="do">## show internal diagnostics boring details</span></span>
<span id="cb9-4"><a href="#cb9-4" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">rba_uniprot_proteins_crossref</span>(</span>
<span id="cb9-5"><a href="#cb9-5" tabindex="-1"></a>  <span class="at">db_id =</span> <span class="st">&quot;CD40&quot;</span>,</span>
<span id="cb9-6"><a href="#cb9-6" tabindex="-1"></a>  <span class="at">db_name =</span> <span class="st">&quot;HGNC&quot;</span>,</span>
<span id="cb9-7"><a href="#cb9-7" tabindex="-1"></a>  <span class="at">diagnostics =</span> <span class="cn">TRUE</span></span>
<span id="cb9-8"><a href="#cb9-8" tabindex="-1"></a>)</span>
<span id="cb9-9"><a href="#cb9-9" tabindex="-1"></a></span>
<span id="cb9-10"><a href="#cb9-10" tabindex="-1"></a><span class="do">## The next function you call, will still use the default rbioapi options</span></span>
<span id="cb9-11"><a href="#cb9-11" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">rba_uniprot_proteins_crossref</span>(</span>
<span id="cb9-12"><a href="#cb9-12" tabindex="-1"></a>  <span class="at">db_id =</span> <span class="st">&quot;CD40&quot;</span>,</span>
<span id="cb9-13"><a href="#cb9-13" tabindex="-1"></a>  <span class="at">db_name =</span> <span class="st">&quot;HGNC&quot;</span></span>
<span id="cb9-14"><a href="#cb9-14" tabindex="-1"></a>)</span></code></pre></div>
<hr />
</div>
</div>
<div id="connection-test" class="section level1 hasAnchor" number="6">
<h1 class="hasAnchor"><span class="header-section-number">6</span>
Connection test<a href="#connection-test" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>The second exception in functions’ naming schema is
<code>rba_connection_test()</code>. Run this simple function to check
your connection with the supported services/databases. If you encounter
errors when using rbioapi, kindly run this function to make sure that
your internet connection or the servers are fine.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a><span class="fu">rba_connection_test</span>(<span class="at">print_output =</span> <span class="cn">TRUE</span>)</span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a><span class="co">#&gt; Checking Your connection to the Databases currently supported by rbioapi:</span></span>
<span id="cb10-3"><a href="#cb10-3" tabindex="-1"></a><span class="co">#&gt; ---&gt;&gt;&gt; Internet :</span></span>
<span id="cb10-4"><a href="#cb10-4" tabindex="-1"></a><span class="co">#&gt; +++ Connected to the Internet.</span></span>
<span id="cb10-5"><a href="#cb10-5" tabindex="-1"></a><span class="co">#&gt; ---&gt;&gt;&gt; Enrichr :</span></span>
<span id="cb10-6"><a href="#cb10-6" tabindex="-1"></a><span class="co">#&gt; +++ The server is responding.</span></span>
<span id="cb10-7"><a href="#cb10-7" tabindex="-1"></a><span class="co">#&gt; ---&gt;&gt;&gt; Ensembl :</span></span>
<span id="cb10-8"><a href="#cb10-8" tabindex="-1"></a><span class="co">#&gt; +++ The server is responding.</span></span>
<span id="cb10-9"><a href="#cb10-9" tabindex="-1"></a><span class="co">#&gt; ---&gt;&gt;&gt; JASPAR :</span></span>
<span id="cb10-10"><a href="#cb10-10" tabindex="-1"></a><span class="co">#&gt; +++ The server is responding.</span></span>
<span id="cb10-11"><a href="#cb10-11" tabindex="-1"></a><span class="co">#&gt; ---&gt;&gt;&gt; miEAA :</span></span>
<span id="cb10-12"><a href="#cb10-12" tabindex="-1"></a><span class="co">#&gt; +++ The server is responding.</span></span>
<span id="cb10-13"><a href="#cb10-13" tabindex="-1"></a><span class="co">#&gt; ---&gt;&gt;&gt; PANTHER :</span></span>
<span id="cb10-14"><a href="#cb10-14" tabindex="-1"></a><span class="co">#&gt; +++ The server is responding.</span></span>
<span id="cb10-15"><a href="#cb10-15" tabindex="-1"></a><span class="co">#&gt; ---&gt;&gt;&gt; Reactome Content Service :</span></span>
<span id="cb10-16"><a href="#cb10-16" tabindex="-1"></a><span class="co">#&gt; +++ The server is responding.</span></span>
<span id="cb10-17"><a href="#cb10-17" tabindex="-1"></a><span class="co">#&gt; ---&gt;&gt;&gt; Reactome Analysis Service :</span></span>
<span id="cb10-18"><a href="#cb10-18" tabindex="-1"></a><span class="co">#&gt; +++ The server is responding.</span></span>
<span id="cb10-19"><a href="#cb10-19" tabindex="-1"></a><span class="co">#&gt; ---&gt;&gt;&gt; STRING :</span></span>
<span id="cb10-20"><a href="#cb10-20" tabindex="-1"></a><span class="co">#&gt; +++ The server is responding.</span></span>
<span id="cb10-21"><a href="#cb10-21" tabindex="-1"></a><span class="co">#&gt; ---&gt;&gt;&gt; UniProt :</span></span>
<span id="cb10-22"><a href="#cb10-22" tabindex="-1"></a><span class="co">#&gt; +++ The server is responding.</span></span></code></pre></div>
<hr />
</div>
<div id="multi-pages" class="section level1 hasAnchor" number="7">
<h1 class="hasAnchor"><span class="header-section-number">7</span>
Iterating over paginated results<a href="#multi-pages" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Some API resources will return paginated responses. This is
particularly common in API resources which return potentially very large
responses. In rbioapi, for these cases, there are arguments such as
“page_number” (with default value of 1) and -if the API resource allows-
“page_size”. To save your time, you may use <code>rba_pages()</code>.
This function will iterate over the pages you have specified.</p>
<p>Take rba_uniprot_taxonomy_name as an example. This function allows
you to search taxonomic nodes in <a href="https://www.uniprot.org" title="Universal Protein Resource (UniProt)">UniProt</a>. The response
can potentially have a huge size, so <a href="https://www.uniprot.org" title="Universal Protein Resource (UniProt)">UniProt</a> returns a
paginated response. For example, if we search for nodes that contain
“adenovirus”, there is a large number of hits:</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a>adeno <span class="ot">&lt;-</span> <span class="fu">rba_uniprot_taxonomy_name</span>(</span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a>  <span class="at">name =</span> <span class="st">&quot;adenovirus&quot;</span>,</span>
<span id="cb11-3"><a href="#cb11-3" tabindex="-1"></a>  <span class="at">search_type =</span> <span class="st">&quot;contain&quot;</span>,</span>
<span id="cb11-4"><a href="#cb11-4" tabindex="-1"></a>  <span class="at">page_number =</span> <span class="dv">1</span></span>
<span id="cb11-5"><a href="#cb11-5" tabindex="-1"></a>)</span>
<span id="cb11-6"><a href="#cb11-6" tabindex="-1"></a></span>
<span id="cb11-7"><a href="#cb11-7" tabindex="-1"></a><span class="fu">str</span>(adeno, <span class="at">max.level =</span> <span class="dv">2</span>)</span>
<span id="cb11-8"><a href="#cb11-8" tabindex="-1"></a><span class="co">#&gt; List of 2</span></span>
<span id="cb11-9"><a href="#cb11-9" tabindex="-1"></a><span class="co">#&gt;  $ taxonomies:&#39;data.frame&#39;:  200 obs. of  8 variables:</span></span>
<span id="cb11-10"><a href="#cb11-10" tabindex="-1"></a><span class="co">#&gt;   ..$ taxonomyId    : int [1:200] 10509 10510 10511 10512 10513 10514 10515 10519 10521 10522 ...</span></span>
<span id="cb11-11"><a href="#cb11-11" tabindex="-1"></a><span class="co">#&gt;   ..$ mnemonic      : chr [1:200] &quot;9ADEN&quot; &quot;ADEB3&quot; &quot;ADEB7&quot; &quot;9ADEN&quot; ...</span></span>
<span id="cb11-12"><a href="#cb11-12" tabindex="-1"></a><span class="co">#&gt;   ..$ scientificName: chr [1:200] &quot;Mastadenovirus&quot; &quot;Bovine adenovirus B serotype 3&quot; &quot;Bovine adenovirus 7&quot; &quot;Canine adenovirus 1&quot; ...</span></span>
<span id="cb11-13"><a href="#cb11-13" tabindex="-1"></a><span class="co">#&gt;   ..$ rank          : chr [1:200] &quot;genus&quot; &quot;serotype&quot; &quot;serotype&quot; &quot;serotype&quot; ...</span></span>
<span id="cb11-14"><a href="#cb11-14" tabindex="-1"></a><span class="co">#&gt;   ..$ superregnum   : chr [1:200] &quot;V&quot; &quot;V&quot; &quot;V&quot; &quot;V&quot; ...</span></span>
<span id="cb11-15"><a href="#cb11-15" tabindex="-1"></a><span class="co">#&gt;   ..$ hidden        : logi [1:200] FALSE TRUE TRUE TRUE TRUE TRUE ...</span></span>
<span id="cb11-16"><a href="#cb11-16" tabindex="-1"></a><span class="co">#&gt;   ..$ commonName    : chr [1:200] NA &quot;BAdV-3&quot; &quot;BAdV-7&quot; NA ...</span></span>
<span id="cb11-17"><a href="#cb11-17" tabindex="-1"></a><span class="co">#&gt;   ..$ synonym       : chr [1:200] NA &quot;Mastadenovirus bos3&quot; NA NA ...</span></span>
<span id="cb11-18"><a href="#cb11-18" tabindex="-1"></a><span class="co">#&gt;  $ pageInfo  :List of 3</span></span>
<span id="cb11-19"><a href="#cb11-19" tabindex="-1"></a><span class="co">#&gt;   ..$ resultsPerPage: int 200</span></span>
<span id="cb11-20"><a href="#cb11-20" tabindex="-1"></a><span class="co">#&gt;   ..$ currentPage   : int 1</span></span>
<span id="cb11-21"><a href="#cb11-21" tabindex="-1"></a><span class="co">#&gt;   ..$ totalRecords  : int 1121</span></span></code></pre></div>
<p>As you can see, the server has returned the first page of the
response, to retrieve the other pages, you should make separate calls
and change the “page_number” argument within each call, or simply use
<code>rba_pages()</code> as demonstrated below:</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a>adeno_pages <span class="ot">=</span> <span class="fu">rba_pages</span>(</span>
<span id="cb12-2"><a href="#cb12-2" tabindex="-1"></a>  <span class="fu">quote</span>(</span>
<span id="cb12-3"><a href="#cb12-3" tabindex="-1"></a>    <span class="fu">rba_uniprot_taxonomy_name</span>(</span>
<span id="cb12-4"><a href="#cb12-4" tabindex="-1"></a>      <span class="at">name =</span> <span class="st">&quot;adenovirus&quot;</span>,</span>
<span id="cb12-5"><a href="#cb12-5" tabindex="-1"></a>      <span class="at">search_type =</span> <span class="st">&quot;contain&quot;</span>,</span>
<span id="cb12-6"><a href="#cb12-6" tabindex="-1"></a>      <span class="at">page_number =</span> <span class="st">&quot;pages:1:3&quot;</span></span>
<span id="cb12-7"><a href="#cb12-7" tabindex="-1"></a>    )</span>
<span id="cb12-8"><a href="#cb12-8" tabindex="-1"></a>  )</span>
<span id="cb12-9"><a href="#cb12-9" tabindex="-1"></a>)</span>
<span id="cb12-10"><a href="#cb12-10" tabindex="-1"></a></span>
<span id="cb12-11"><a href="#cb12-11" tabindex="-1"></a><span class="do">## You can inspect the structure of the response:</span></span>
<span id="cb12-12"><a href="#cb12-12" tabindex="-1"></a><span class="fu">str</span>(adeno_pages, <span class="at">max.level =</span> <span class="dv">2</span>)</span>
<span id="cb12-13"><a href="#cb12-13" tabindex="-1"></a><span class="co">#&gt; List of 3</span></span>
<span id="cb12-14"><a href="#cb12-14" tabindex="-1"></a><span class="co">#&gt;  $ page_1:List of 2</span></span>
<span id="cb12-15"><a href="#cb12-15" tabindex="-1"></a><span class="co">#&gt;   ..$ taxonomies:&#39;data.frame&#39;:   200 obs. of  8 variables:</span></span>
<span id="cb12-16"><a href="#cb12-16" tabindex="-1"></a><span class="co">#&gt;   ..$ pageInfo  :List of 3</span></span>
<span id="cb12-17"><a href="#cb12-17" tabindex="-1"></a><span class="co">#&gt;  $ page_2:List of 2</span></span>
<span id="cb12-18"><a href="#cb12-18" tabindex="-1"></a><span class="co">#&gt;   ..$ taxonomies:&#39;data.frame&#39;:   200 obs. of  8 variables:</span></span>
<span id="cb12-19"><a href="#cb12-19" tabindex="-1"></a><span class="co">#&gt;   ..$ pageInfo  :List of 3</span></span>
<span id="cb12-20"><a href="#cb12-20" tabindex="-1"></a><span class="co">#&gt;  $ page_3:List of 2</span></span>
<span id="cb12-21"><a href="#cb12-21" tabindex="-1"></a><span class="co">#&gt;   ..$ taxonomies:&#39;data.frame&#39;:   200 obs. of  6 variables:</span></span>
<span id="cb12-22"><a href="#cb12-22" tabindex="-1"></a><span class="co">#&gt;   ..$ pageInfo  :List of 3</span></span></code></pre></div>
<p>As you can see, what we have done was:</p>
<ol style="list-style-type: decimal">
<li><p>Wrap the function call in <code>qoute()</code> and enter that as
the input for <code>rba_pages()</code>.</p></li>
<li><p>Replace the argument we want to iterate over it, with a string in
this format: “pages:start:end”. For example, we supplied page_number =
“pages:1:3” to get the responses of pages 1 to 3.</p></li>
</ol>
<hr />
</div>
<div id="how-to-cite" class="section level1 hasAnchor" number="8">
<h1 class="hasAnchor"><span class="header-section-number">8</span> How
and what to cite?<a href="#how-to-cite" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>rbioapi is an interface between you and other databases and services.
Thus, if you have used rbioapi in published research, <strong>in
addition to kindly citing rbioapi, <u><em>ensure to fully and properly
cite the databases/services you have used</em></u></strong>. Suggested
citations have been added in the functions’ manuals, under the
“references” section; Nevertheless, it is the user’s responsibility to
check for proper citations and to properly cite the database/services
that they have used.</p>
<div id="how-to-cite-rbioapi" class="section level2 hasAnchor" number="8.1">
<h2 class="hasAnchor"><span class="header-section-number">8.1</span> How
to cite rbioapi<a href="#how-to-cite-rbioapi" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li>Moosa Rezwani, Ali Akbar Pourfathollah, Farshid Noorbakhsh, rbioapi:
user-friendly R interface to biologic web services’ API, Bioinformatics,
Volume 38, Issue 10, 15 May 2022, Pages 2952–2953, <a href="https://doi.org/10.1093/bioinformatics/btac172" class="uri">https://doi.org/10.1093/bioinformatics/btac172</a></li>
</ul>
</div>
<div id="how-to-cite-services" class="section level2 hasAnchor" number="8.2">
<h2 class="hasAnchor"><span class="header-section-number">8.2</span> How
to cite the databases and web services<a href="#how-to-cite-services" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li><p><a href="rbioapi_enrichr.html#citations" title="How to cite Enrichr">How to cite Enrichr</a>. (See on <a href="https://maayanlab.cloud/Enrichr/help#terms">Enrichr
website</a>)</p></li>
<li><p><a href="rbioapi_jaspar.html#citations" title="How to cite JASPAR">How to cite JASPAR</a>. (See on <a href="https://jaspar.elixir.no/faq/">JASPAR website</a>)</p></li>
<li><p><a href="rbioapi_mieaa.html#citations" title="How to cite miEAA">How to cite miEAA</a>. (See on <a href="https://ccb-compute2.cs.uni-saarland.de/mieaa2">miEAA
website</a>)</p></li>
<li><p><a href="rbioapi_panther.html#citations" title="How to cite PANTHER">How to cite PANTHER</a>. (See on <a href="https://www.pantherdb.org/publications.jsp#HowToCitePANTHER">PANTHER
website</a>)</p></li>
<li><p><a href="rbioapi_reactome.html#citations" title="How to cite Reactome">How to cite Reactome</a>. (See on <a href="https://reactome.org/cite">Reactome website</a>)</p></li>
<li><p><a href="rbioapi_string.html#citations" title="How to cite STRING">How to cite STRING</a>. (See on <a href="https://string-db.org/cgi/about?footer_active_subpage=references">STRING
website</a>)</p></li>
<li><p><a href="rbioapi_uniprot.html#citations" title="How to cite UniProt">How to cite UniProt</a>. (See on <a href="https://www.uniprot.org/help/publications">UniProt
website</a>)</p></li>
</ul>
<hr />
</div>
</div>
<div id="code-of-conduct" class="section level1 hasAnchor" number="9">
<h1 class="hasAnchor"><span class="header-section-number">9</span> Code
of conduct<a href="#code-of-conduct" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>This package, rbioapi, is an unofficial interface implementation and
is not associated, endorsed, or officially connected in any way with the
original databases and web services. The creators and maintainers of
rbioapi are independent entities and have no official relationship with
those databases and web services.</p>
<p>When using rbioapi, remember that you are querying data from web
services; So please be considerate. Never flood a server with requests,
if you need to download <em>unreasonably</em> large volumes of data,
directly downloading the databases supplied in those services may be a
better alternative. If you see yourself being rate-limited from any
server (HTTP <strong>429 Too Many Requests</strong> response status
code), know that you are sending more requests than what the server
interprets as normal behavior, so please seek other methods or use
<code>Sys.sleep()</code> between your requests.</p>
<hr />
</div>
<div id="what-next" class="section level1 hasAnchor" number="10">
<h1 class="hasAnchor"><span class="header-section-number">10</span> What
next?<a href="#what-next" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Each supported service has a dedicated vignette article. Make sure to
check those too.</p>
<ol style="list-style-type: decimal">
<li><a href="rbioapi_enrichr.html" title="rbioapi &amp; Enrichr vignette">Enrichr</a> <sup>(<a href="https://rbioapi.moosa-r.com/articles/rbioapi_enrichr.html" title="rbioapi &amp; Enrichr vignette">Documentation site</a>)</sup></li>
<li><a href="rbioapi_jaspar.html" title="rbioapi &amp; JASPAR vignette article">JASPAR</a> <sup>(<a href="https://rbioapi.moosa-r.com/articles/rbioapi_jaspar.html" title="rbioapi &amp; JASPAR vignette article">Documentation site</a>)</sup></li>
<li><a href="rbioapi_mieaa.html" title="rbioapi &amp; miEAA vignette article">miEAA</a> <sup>(<a href="https://rbioapi.moosa-r.com/articles/rbioapi_mieaa.html" title="rbioapi &amp; miEAA vignette article">Documentation site</a>)</sup></li>
<li><a href="rbioapi_panther.html" title="rbioapi &amp; PANTHER vignette article">PANTHER</a> <sup>(<a href="https://rbioapi.moosa-r.com/articles/rbioapi_panther.html" title="rbioapi &amp; PANTHER vignette article">Documentation site</a>)</sup></li>
<li><a href="rbioapi_reactome.html" title="rbioapi &amp; Reactome vignette article">Reactome</a> <sup>(<a href="https://rbioapi.moosa-r.com/articles/rbioapi_reactome.html" title="rbioapi &amp; Reactome vignette article">Documentation site</a>)</sup></li>
<li><a href="rbioapi_string.html" title="rbioapi &amp; STRING vignette article">STRING</a> <sup>(<a href="https://rbioapi.moosa-r.com/articles/rbioapi_string.html" title="rbioapi &amp; STRING vignette article">Documentation site</a>)</sup></li>
<li><a href="rbioapi_uniprot.html" title="rbioapi &amp; UniProt vignette article">UniProt</a> <sup>(<a href="https://rbioapi.moosa-r.com/articles/rbioapi_uniprot.html" title="rbioapi &amp; UniProt vignette article">Documentation site</a>)</sup></li>
</ol>
<p>We are also adding vignette articles focusing on tasks and
workflows:</p>
<ol style="list-style-type: decimal">
<li><a href="rbioapi_do_enrich.html" title="Do with rbioapi: Enrichment (Over-Representation) Analysis in R">Do
with rbioapi: Enrichment (Over-Representation) Analysis in R</a>
<sup>(<a href="https://rbioapi.moosa-r.com/articles/rbioapi_do_enrich.html" title="rbioapi &amp; UniProt vignette article">documentation site</a>)</sup></li>
</ol>
<hr />
</div>
<div id="design" class="section level1 hasAnchor" number="11">
<h1 class="hasAnchor"><span class="header-section-number">11</span>
Design of rbioapi<a href="#design" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>To learn more about the design and concepts behind developing
rbioapi, please read <a href="https://doi.org/10.1093/bioinformatics/btac172" title="Rezwani, M., Pourfathollah, A. A., &amp; Noorbakhsh, F. (2022). rbioapi: user-friendly R interface to biologic web services’ API. Bioinformatics, 38(10), 2952–2953. doi: 10.1093/bioinformatics/btac172">our
paper in Bioinformatics</a>.</p>
<hr />
</div>
<div id="links" class="section level1 hasAnchor" number="12">
<h1 class="hasAnchor"><span class="header-section-number">12</span>
Links<a href="#links" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<ul>
<li><p><a href="https://rbioapi.moosa-r.com/articles/rbioapi.html" title="rbioapi: User-Friendly R Interface to Biologic Web Services&#39; API">This
article in rbioapi documentation site</a></p></li>
<li><p><a href="https://rbioapi.moosa-r.com/reference/index.html" title="rbioapi reference">Functions references in rbioapi documentation
site</a></p></li>
</ul>
<hr />
</div>
<div id="session-info" class="section level1 hasAnchor" number="13">
<h1 class="hasAnchor"><span class="header-section-number">13</span>
Session info<a href="#session-info" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<pre><code>#&gt; R version 4.5.1 (2025-06-13 ucrt)
#&gt; Platform: x86_64-w64-mingw32/x64
#&gt; Running under: Windows 11 x64 (build 26100)
#&gt; 
#&gt; Matrix products: default
#&gt;   LAPACK version 3.12.1
#&gt; 
#&gt; locale:
#&gt; [1] LC_COLLATE=C                          
#&gt; [2] LC_CTYPE=English_United States.utf8   
#&gt; [3] LC_MONETARY=English_United States.utf8
#&gt; [4] LC_NUMERIC=C                          
#&gt; [5] LC_TIME=English_United States.utf8    
#&gt; 
#&gt; time zone: Europe/Brussels
#&gt; tzcode source: internal
#&gt; 
#&gt; attached base packages:
#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     
#&gt; 
#&gt; other attached packages:
#&gt; [1] rbioapi_0.8.3
#&gt; 
#&gt; loaded via a namespace (and not attached):
#&gt;  [1] digest_0.6.37     R6_2.6.1          fastmap_1.2.0     xfun_0.52        
#&gt;  [5] cachem_1.1.0      knitr_1.50        htmltools_0.5.8.1 rmarkdown_2.29   
#&gt;  [9] lifecycle_1.0.4   cli_3.6.5         sass_0.4.10       jquerylib_0.1.4  
#&gt; [13] compiler_4.5.1    httr_1.4.7        rstudioapi_0.17.1 tools_4.5.1      
#&gt; [17] curl_6.4.0        evaluate_1.0.4    bslib_0.9.0       yaml_2.3.10      
#&gt; [21] rlang_1.1.6       jsonlite_2.0.0</code></pre>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
